* TOC
{:toc}

## Biotech 7005/Bioinf 3000: Frontiers of Biotechnology: Bioinformatics and Systems Modelling
{:.no_toc}

Semester 2 2022

Coordinator:
- Dave Adelson

Instructors:
- Dave Adelson
- Steven Delean
- Anna Sheppard
- Bastien Llamas
- Zhipeng Qu
- David Lawrence

Teaching assistants:
- Wenjun (Nora) Liu
- Alex Stuart

__*Course Contact*__
Dave Adelson
- Email: david.adelson@adelaide.edu.au
- Phone: 08 8313 7555
- Office: Molecular Life Sciences Rm 2.10

### Contact Hours
Lectures, Tutorials and practicals will be delivered face to face (subject to change depending on COVID-19). Social distancing and or mask wearing may be required. 

Lectures: Monday 12 Noon, Lower Napier, LG28
- 25 July — 24 Oct

Tutorials: Tuesday 11am-1pm, Mawson, 134
- 26 July — 25 Oct

Practicals: Friday 11am-1pm, Johnson, 111
- 29 July — 28 Oct

### Timetable (Provisional)

| **Week** | **Monday** |**Lecture**         |**Tutorial**                                 | **Practical**                              |
|----------|------------|---------------------|--------------------------------------------|------------------------------------------|
| **1**    | 25/7       |Intro to Bioinformatics (Dave)  |[Introduction to VMs]/[Introduction to R] (1)   |[Introduction to R] (1.5) (Steven/Dave)  |
| **2**    | 1/8        |[Statistics][2] (Steven)  |[Statistics worksheet]  (Steven/Dave) |[Introduction to R] (2) (Steven/Nora)      |
| **3**    | 8/8        |Evolutionary Processes (Dave) |*Stats Test* (Steven/Dave)  |[Introduction To Bash] (Dave/Alex)   |
| **4**    | 15/8       |Next-generation sequencing  (Dave) |*Evolution test* (Dave)|[Scripting In Bash] (Dave/Alex)                   |
| **5**    | 22/8       |Alignment Process (Dave) |*NGS test* <br> Scripting tutorial (Anna) | [Evolutionary processes] (Dave/Alex)    |
| **6**    | 29/8       |Alignment Process (Dave) |*No test* (Dave)/Alignment tutorial  |[Alignment/NGS (1)] (Anna/Alex)             |
| **7**    | 5/9        |Reproducible Science (David L)   |*Alignment test* (Dave)/ [Coding comments] (David L)  |[Alignment/NGS (2)] (Dave/Alex)             |
| **8**    | 12/9       |Version Control (David L)   |[Version control tutorial] (David L)    |[Local Version Control] (David L/Nora)  |
| **-**    | 19/9       | Mid-Semester Break      |                                        |                                          |
| **-**    | 26/9       | Mid-Semester Break      |                                        |                                          |
| **9**    | 3/10 (PH)  | *Public Holiday 3/10* No lecture|*Version control test*/ GitHub tutorial (David L)  |GitHub Repositories (David L/Nora)  
| **10**   | 10/10      | Quantifying Gene Expression (Zhipeng)|[DE gene tutorial] (Zhipeng) |[Transcriptomic Practical] (1) (Zhipeng/Nora)  |
| **11**   | 17/10      | Transcriptomics (Zhipeng) |Transcriptome tutorial (Zhipeng) |[Transcriptomic Practical] (2) (Zhipeng/Nora)  | 
| **12**   | 24/10      | Ancient DNA (Bastien)|*Transcriptomic/Gene Expression test*/ aDNA tutorial |Can work on last assessment |


[1]: http://biotech7005.services.adelaide.edu.au/01-bioinformatics.slide
[2]: Lectures/02-statistics.html
[3]: http://biotech7005.services.adelaide.edu.au/03-evoprocess.slide
[4]: http://biotech7005.services.adelaide.edu.au/04-sequencing.slide
[5]: http://biotech7005.services.adelaide.edu.au/05-alignment.slide
[6]: http://biotech7005.services.adelaide.edu.au/05-alignment.slide
[7]: https://university-of-adelaide-bx-masters.github.io/BIOTECH-7005-BIOINF-3000/
[8]: https://university-of-adelaide-bx-masters.github.io/BIOTECH-7005-BIOINF-3000/
[9]: Practicals/VCF_Analysis/Week_8_Practical-VCF_Analysis.md
[10]: Practicals/Transcriptome_Practical/Transcriptome_assembly.md


[Introduction to VMs]: Practicals/vm_connection.md
[Introduction to R]: Practicals/R_Practicals
[Introduction To Bash]: Practicals/Bash_Practicals/1_IntroBash.md
[Scripting In Bash]: Practicals/Bash_Practicals/2_BashScripting.md
[Evolutionary processes]: Practicals/evolutionary_prac/evolutionary.md
[Alignment/NGS (1)]: Practicals/NGS_Practicals/1_NGS_Practical1.md
[Alignment/NGS (2)]: Practicals/NGS_Practicals/2_NGS_Practical2.md
[Variant Calling practical]: Practicals/VCF_Analysis/Week_8_Practical-VCF_Analysis.md
[Graphical analyses]: https://github.com/kortschak/graphprac/
[DE gene tutorial]: DE_gene_tutorial/Tutorial_DE_Genes.html
[Statistics worksheet]: Tutorials/Wk2_Statistics.html
[Transcriptomic Practical]: Practicals/Transcriptome_Practical/Transcriptome_assembly.md
[Coding comments]: Tutorials/Code_comments_tutorial/code_comments.html
[Version control tutorial]: Tutorials/Version_control_tutorial/version_control_tutorial.html
[Local Version Control]: Practicals/local_version_control_practical/local_version_control_practical.rmd

## Assessment

### Assessment Schedule & Weighting

| **Assessment - All Students**                                 | **Subject**                |  **UG Weight**  | **PG Weight** | **Due Date**          |
|--------------------------------------------------------------|-----------------------------|-----------------|------------|-----------------------|
| [Assessment 1](Assignments/Assignment1.md)                   | R programming               |  15%             | N/A%         | Friday 12/08/22     |
| [Assessment 2](Assignments/Assignment2.md)                   | Bash Scripting & Statistics |  15%             | N/A%          | Friday 26/08/22     |
| [Assessment 3](Practicals/evolutionary_prac/evolutionary.md) | Evolutionary processes      |  15%             | N/A%         | Friday 02/09/22  |
| [Assessment 4](Assignments/Assignment4.md)                   | Alignment/NGS               |  15%             | N/A%          | Friday 16/09/22  |
| [Assessment 5](Assignments/Assignment5.md)                   | Version Control             |  15%             | N/A%         | Friday 14/10/22  |
| [Assessment 6](Assignments/Assignment6_transcriptome_assembly.md)      | Transcriptomics|  15%             | N/A%         | Friday 28/10/22    |
| Tests                                                        |                             |  10%            | N/A%         |                       |
| [Project](Assignments/major_project.md)                      | Complete Dataset            |  N/A            |N/A%        |    |
| Exam                                                         |                             |  N/A            | N/A%        |                       |

Note that the Project assignment only applies to Masters level students. Undergraduates do not do a Project. 

### Assessment Checklist

Have you:

- [ ] Answered all the questions?
- [ ] Followed naming conventions for Assessments?
- [ ] Attached an assessment coversheet? (Available [here](COVERSHEET.md)).
	- This is a course requirement. See the [Academic Honesty Policy](http://www.adelaide.edu.au/policies/230/).
- [ ] Identified the work as yours?
	- Emails should have the course and assessment task names.
	- Documents should be named with your name, the course name and the assessment task.
	- Printed documents should have you name and the course and assessment task in the text/footer/header.
- [ ] Used appropriate electronic communication with assessors?
	- Emails should have a meaningful subject.
- [ ] Handed in the assignment before the due time (Friday tutorial 9am)?

### Major Project (15%) - Masters Students Only

In this course, the following next-generation sequencing (NGS) datasets/protocols are covered:

1. Whole genome sequencing/Resequencing
2. Transcriptome Sequencing (RNAseq)
3. Variant Calling

Each of these NGS approaches uses similar programs and analysis approaches, such as quality control (quality and sequencing adapter trimming), genome alignment, and downstream visualisation and statistical methods. For the major project, you will take a published dataset and complete all the analysis tasks (from raw data to final results) during the course.

You have the freedom to choose any dataset you would like before we start week 6's practical (Alignment/NGS practical), or choose from datasets that we have chosen. You will be asked assessment questions on your data during practical assessments throughout the semester, and a final report will be due the week before the end of semester (Week 12)

## Useful Links

[How To Ask Questions The Smart Way](http://www.catb.org/esr/faqs/smart-questions.html)

[How to write a good bug report](https://musescore.org/en/developers-handbook/how-write-good-bug-report-step-step-instructions)
