* TOC
{:toc}

## Biotech 7005/Bioinf 3000: Frontiers of Biotechnology: Bioinformatics and Systems Modelling
{:.no_toc}

Semester 2 2023

Coordinator:
- Dave Adelson

Instructors:
- Dave Adelson
- Steven Delean
- Anna Sheppard
- Bastien Llamas
- Zhipeng Qu
- David Lawrence

Teaching assistants:
- TBD
- TBD

__*Course Contact*__
Dave Adelson
- Email: david.adelson@adelaide.edu.au
- Phone: 08 8313 7555
- Office: Molecular Life Sciences Rm 2.10

### Contact Hours
Lectures and practicals will be delivered face to face. If you are unwell, please think of others and do not attend. 

Lectures: Monday 9AM, Napier, G03
- 24 July — 23 Oct

Practical 01: Monday 11am-1pm, Johnson, 111
- 24 July — 23 Oct

Practical 02: Friday 11am-1pm, Johnson, 111
- 28 July — 27 Oct

### Timetable (Provisional)

| **Week** | **Monday** |**Lecture**         |**Monday Practical**                                 | **Friday Practical**                              |
|----------|------------|---------------------|--------------------------------------------|------------------------------------------|
| **1**    | 24/7       |Intro to Bioinformatics (Dave)  |[Introduction to VMs]/[Introduction to R] (1)   |[Introduction to R] (1.5) (Steven/Dave)  |
| **2**    | 31/7        |[Statistics][2] (Steven)  |[Statistics worksheet]  (Steven/Dave) |[Introduction to R] (2) (Steven/TBD)      |
| **3**    | 7/8        |Evolutionary Processes (Dave) |*Stats Test* [Introduction to Bash] (1) (Steven/Anna)  |[Introduction To Bash] (2) (Anna)   |
| **4**    | 14/8       |Next-generation sequencing  (Dave) |*Evolution test* [Scripting in Bash] (1) (Dave)|[Scripting In Bash] (2)(Dave)                   |
| **5**    | 21/8       |Alignment Process (Dave) |*NGS test* [Evolutionary Processes] (1) (Dave)) | [Evolutionary processes] (2) (Dave/)    |
| **6**    | 28/8       |Alignment Process (Dave) |*No test* [Alignment/NGS (1)] (Dave)  |[Alignment/NGS (2)] (Dave)             |
| **7**    | 4/9        |Reproducible Science (David L)   |*Alignment test* (Dave)/ [Coding comments] (David L)  |[Version Control] (1) David L             |
| **8**    | 11/9       |Version Control (David L)   |[Version Control] (2) (David L)    |[Local Version Control] (David L)  |
| **-**    | 18/9       | Mid-Semester Break      |                                        |                                          |
| **-**    | 25/9       | Mid-Semester Break      |                                        |                                          |
| **9**    | 2/10 (PH)  | *Public Holiday 3/10* No lecture|*Public Holiday No practical*  | *Version control test*/ [GitHub Intro] (David L)   
| **10**   | 9/10      | Quantifying Gene Expression (Zhipeng)|[GitHub Repositories] (David L) | [DE genes] (Zhipeng)  |
| **11**   | 16/10      | Transcriptomics (Zhipeng) | [Transcriptome intro] (Zhipeng) |*Transcriptomic/Gene Expression test*/ [Transcriptomic Practical][10] (1) (Zhipeng)  | 
| **12**   | 23/10      |[Ancient DNA][11] (Bastien)|[Transcriptomic Practical][10] (2) (Zhipeng) | [aDNA][12] |


[1]: http://biotech7005.services.adelaide.edu.au/01-bioinformatics.slide
[2]: Lectures/02-statistics.html
[3]: http://biotech7005.services.adelaide.edu.au/03-evoprocess.slide
[4]: http://biotech7005.services.adelaide.edu.au/04-sequencing.slide
[5]: http://biotech7005.services.adelaide.edu.au/05-alignment.slide
[6]: http://biotech7005.services.adelaide.edu.au/05-alignment.slide
[7]: https://university-of-adelaide-bx-masters.github.io/BIOTECH-7005-BIOINF-3000/
[8]: https://university-of-adelaide-bx-masters.github.io/BIOTECH-7005-BIOINF-3000/
[9]: Practicals/VCF_Analysis/Week_8_Practical-VCF_Analysis.md
[10]: Practicals/Transcriptome_Practical/Transcriptome_assembly.md
[11]: https://github.com/University-of-Adelaide-Bx-Masters/BIOTECH-7005-BIOINF-3000/blob/master/Lectures/12_ancient_DNA-compressed.pdf
[12]: https://github.com/University-of-Adelaide-Bx-Masters/BIOTECH-7005-BIOINF-3000/tree/master/Practicals/ancient_DNA_practical


[Introduction to VMs]: Practicals/vm_connection.md
[Introduction to R]: Practicals/R_Practicals
[Introduction To Bash]: Practicals/Bash_Practicals/1_IntroBash.md
[Scripting In Bash]: Practicals/Bash_Practicals/2_BashScripting.md
[Evolutionary processes]: Practicals/evolutionary_prac/evolutionary.md
[Alignment/NGS (1)]: Practicals/NGS_Practicals/1_NGS_Practical1.md
[Alignment/NGS (2)]: Practicals/NGS_Practicals/2_NGS_Practical2.md
[Variant Calling practical]: Practicals/VCF_Analysis/Week_8_Practical-VCF_Analysis.md
[Graphical analyses]: https://github.com/kortschak/graphprac/
[DE gene tutorial]: Tutorials/Tutorial10_DE_analysis.pdf 
[Statistics worksheet]: Tutorials/Wk2_Statistics.html
[Coding comments]: Practicals/Code_comments_practical/code_comments.html
[Version Control]: Practicals/Version_control_practical/version_control_practical.html
[Local Version Control]: Practicals/local_version_control_practical/local_version_control_practical.html
[GitHub repositories]: Practicals/github_repositories_practical/github_repositories_practical.html

## Assessment

### Assessment Schedule & Weighting (NOT YET UPDATED FOR 2023)

| **Assessment - All Students**                                 | **Subject**                |  **UG Weight**  | **PG Weight** | **Due Date**          |
|--------------------------------------------------------------|-----------------------------|-----------------|------------|-----------------------|
| [Assessment 1](Assignments/Assignment1.md)                   | R programming               |  15%            | 12.75%         | Friday 11/08/23     |
| [Assessment 2](Assignments/Assignment2.md)                   | Bash Scripting & Statistics |  15%            | 12.75%         | Friday 25/08/23     |
| [Assessment 3](Practicals/evolutionary_prac/evolutionary.md) | Evolutionary processes      |  15%            | 12.75%         | Friday 01/09/23  |
| [Assessment 4](Assignments/Assignment4.md)                   | Alignment/NGS               |  15%            | 12.75%         | Friday 15/09/23  |
| [Assessment 5](Assignments/Assignment5.md)                   | Version Control             |  15%            | 12.75%         | Friday 13/10/23  |
| [Assessment 6](Assignments/Assignment6_transcriptome_assembly.md)      | Transcriptomics   |  15%            | 12.75%         | Friday 27/10/23    |
| Tests (x6)                                                   |                             |  10%            | 8.5%           |                       |
| [Project]THIS IS UNDER CONSTRUCTION (Assignments/major_project.md)                      | Complete Dataset            |  N/A            | 15%            |  Friday 03/11/23  |


Note that the Project assignment only applies to Masters level students. Undergraduates do not do a Project. 

### Assessment Checklist

Have you:

- [ ] Answered all the questions?
- [ ] Followed naming conventions for Assessments?
- [ ] Attached an assessment coversheet? (Available [here](COVERSHEET.md)).
	- This is a course requirement. See the [Academic Honesty Policy](http://www.adelaide.edu.au/policies/230/).
- [ ] Identified the work as yours?
	- Emails should have the course and assessment task names.
	- Documents should be named with your name, the course name and the assessment task.
	- Printed documents should have you name and the course and assessment task in the text/footer/header.
- [ ] Used appropriate electronic communication with assessors?
	- Emails should have a meaningful subject.
- [ ] Handed in the assignment before the due time (Friday tutorial 9am)?

### Major Project (15%) - Masters Students Only

In this course, the following next-generation sequencing (NGS) datasets/protocols are covered:

1. Phylogenetic trees
2. Whole genome resequencing 
3. Transcriptome Sequencing (RNAseq)  

Each of these NGS approaches uses similar programs and analysis approaches, such as quality control (quality and sequencing adapter trimming), genome alignment, and downstream visualisation and statistical methods. For the major project, you will take a published dataset and complete all the analysis tasks (from raw data to final results) during the course.

You have the freedom to choose any dataset you would like before we start week 6's practical (Alignment/NGS practical), or choose from datasets that we have chosen. You will be asked assessment questions on your data during practical assessments throughout the semester, and a final report will be due the week before the end of semester (Week 12)

## Useful Links

[How To Ask Questions The Smart Way](http://www.catb.org/esr/faqs/smart-questions.html)

[How to write a good bug report](https://musescore.org/en/developers-handbook/how-write-good-bug-report-step-step-instructions)
