<!DOCTYPE html>
<html lang="en"><head>
<script src="transcriptomics-lecture1-slides_files/libs/clipboard/clipboard.min.js"></script>
<script src="transcriptomics-lecture1-slides_files/libs/quarto-html/tabby.min.js"></script>
<script src="transcriptomics-lecture1-slides_files/libs/quarto-html/popper.min.js"></script>
<script src="transcriptomics-lecture1-slides_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="transcriptomics-lecture1-slides_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="transcriptomics-lecture1-slides_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="transcriptomics-lecture1-slides_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.7.32">

  <meta name="author" content=" Dr Stevie Pederson (They/Them)">
  <meta name="dcterms.date" content="2025-09-01">
  <title>Transcriptomics: Lecture 1</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="transcriptomics-lecture1-slides_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="transcriptomics-lecture1-slides_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for citations */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
      margin-bottom: 0em;
    }
    .hanging-indent div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }  </style>
  <link rel="stylesheet" href="transcriptomics-lecture1-slides_files/libs/revealjs/dist/theme/quarto-6a216ba96c5c25e91d68b44b6d1a2940.css">
  <link href="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Transcriptomics: Lecture 1</h1>
  <p class="subtitle"></p><div class="line-block">Biotech 7005/Bioinf 3000<br>
Frontiers of Biotechnology: Bioinformatics and Systems Modelling<br>
The University of Adelaide</div><p></p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
<br>Dr Stevie Pederson (They/Them) 
</div>
        <p class="quarto-title-affiliation">
            </p><div class="line-block">Black Ochre Data Labs, The Kids Research Institute Australia</div>
          <p></p>
    </div>
</div>

  <p class="date">September 1, 2025</p>
</section>
<section id="welcome-to-country" class="slide level2">
<h2><span class="text-red">Welcome To Country</span></h2>
<div class="text-red">
<p><em>I’d like to acknowledge the Kaurna people as the traditional owners and custodians of the land we know today as the Adelaide Plains, where I live &amp; work.</em> <br><br> <em>I also acknowledge the deep feelings of attachment and relationship of the Kaurna people to their place.</em> <br><br> <em>I pay my respects to the cultural authority of Aboriginal and Torres Strait Islander peoples from other areas of Australia, and pay our respects to Elders past, present and emerging, and acknowledge any Aboriginal Australians who may be with us today</em></p>
</div>
</section>
<section>
<section id="introduction-to-transcriptomics" class="title-slide slide level1 center">
<h1>Introduction To Transcriptomics</h1>

</section>
<section id="introduction" class="slide level2">
<h2>Introduction</h2>
<ul>
<li>Postdoctoral Bioinformatician, Black Ochre Data Labs, Adelaide</li>
<li>Working in collaboration with members of the SA Aboriginal community</li>
<li>Multi-omics project to identify and address the underlying causes of high T2D rates and complications
<ul>
<li>Using genomics, epigenomics, transcriptomics and other layers</li>
<li>My focus is on the transcriptomics layer</li>
</ul></li>
</ul>
</section>
<section id="why-transcriptomics" class="slide level2">
<h2>Why Transcriptomics?</h2>
<div class="columns">
<div class="column">
<ul>
<li>DNA can be described as being like a giant book of instructions</li>
<li>Some regions are defined as <em>genes</em>
<ul>
<li>Originally considered to be the basic unit of inheritance</li>
<li>Now commonly used to describe a region of DNA transcribed into RNA</li>
</ul></li>
</ul>
</div><div class="column">
<div class="fragment">
<div style="font-size: 60%; text-align: center;">
<p><img data-src="transcriptomics/DNA_to_protein_or_ncRNA.png"> <br>By <a href="//commons.wikimedia.org/wiki/User:Evolution_and_evolvability" title="User:Evolution and evolvability">Thomas Shafee</a> - <span class="int-own-work" lang="en">Own work</span>, <a href="https://creativecommons.org/licenses/by/4.0" title="Creative Commons Attribution 4.0">CC BY 4.0</a>, <a href="https://commons.wikimedia.org/w/index.php?curid=39441809">Wikimedia Link</a></p>
</div>
</div>
</div></div>
</section>
<section id="why-transcriptomics-1" class="slide level2 slide-only unlisted">
<h2>Why Transcriptomics?</h2>
<ul>
<li>DNA <span class="math inline">\(\rightarrow\)</span> mRNA <span class="math inline">\(\rightarrow\)</span> Proteins
<ul>
<li>Commonly referred to as the Central Dogma of Biology</li>
</ul></li>
<li>Proteins are the <em>workhorses</em> of the cell &amp; body
<ul>
<li>Do most of the work, and are responsible for most of the structure</li>
<li>Examples like keratin (hair), haemoglobin (oxygen transport) etc</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>ncRNAs are also highly functional
<ul>
<li>Ribosomal RNA (rRNA) essential for translation from mRNA to Protein</li>
<li>microRNAs play a role in gene-regulation via mRNA stability</li>
<li>The lncRNA Xist coats the <em>entire X chromosome</em> during X inactivation</li>
</ul></li>
</ul>
</div>
</section>
<section id="why-transcriptomics-2" class="slide level2 slide-only unlisted">
<h2>Why Transcriptomics?</h2>
<h3 id="definition">Definition</h3>
<p>Based on <span class="citation" data-cites="Wang2009-hf">Wang, Gerstein, and Snyder (<a href="#/references" role="doc-biblioref" onclick="">2009</a>)</span></p>
<blockquote>
<p>The transcriptome can be defined as the complete set of (RNA) transcripts in a cell, or a population of cells, for a specific developmental stage or physiological condition</p>
</blockquote>
<div class="fragment">
<ul>
<li>Transcriptomics is simply the study of the transcriptome</li>
<li>Can be the <em>entire RNA content of a cell</em> (or cells) or a subset of molecules (e.g.&nbsp;mRNA, miRNA)</li>
</ul>
</div>
</section>
<section id="why-transcriptomics-3" class="slide level2 slide-only unlisted">
<h2>Why Transcriptomics?</h2>
<div class="columns">
<div class="column">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="transcriptomics/pgen.1005668.g006.png"></p>
<figcaption>Taken from <span class="citation" data-cites="Fang2015-uq">Fang et al. (<a href="#/references" role="doc-biblioref" onclick="">2015</a>)</span></figcaption>
</figure>
</div>
</div><div class="column">
<ul>
<li>Most RNA is single-stranded but can have extremely complex structure
<ul>
<li>Shown is a 2kb region from the lncRNA Xist (17kb in total)</li>
</ul></li>
<li>Also interacts with the antisense lncRNA Tsix</li>
</ul>
</div></div>
</section>
<section id="why-transcriptomics-4" class="slide level2 slide-only unlisted">
<h2>Why Transcriptomics?</h2>
<ul>
<li>Is a snapshot of the <strong>dynamic biological processes</strong> associated with a biological question</li>
<li>Use to make inference about these processes
<ul>
<li>Identify therapeutic targets for Cardiovascular Disease</li>
<li>Biomarkers for CAR-T cells</li>
<li>Key drivers of correlated gene networks</li>
<li>Early drivers of neurodegeneration in Alzheimers</li>
</ul></li>
<li>Assumed to be low-level
<ul>
<li>DNA <span class="math inline">\(\rightarrow\)</span> <span class="text-red">RNA</span> <span class="fragment"><span class="math inline">\(\rightarrow\)</span> Protein <span class="math inline">\(\rightarrow\)</span> Metabolites, Signalling molecules, etc …</span></li>
</ul></li>
</ul>
</section>
<section id="why-transcriptomics-5" class="slide level2 slide-only unlisted">
<h2>Why Transcriptomics?</h2>
<ul>
<li>Is the first molecular level where quantity becomes a key aspect
<ul>
<li>Highly-expressed, or low-expressed genes are important</li>
<li>Changes in response to stimulus impact gene expression levels</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Much of the early transcriptomic analyses were quantitative
<ul>
<li>Sequence variation often captured at DNA-level</li>
</ul></li>
<li>Now extending to transcript structure and modifications
<ul>
<li>Identification of fusion transcripts, RNA-methylation etc</li>
</ul></li>
</ul>
</div>
</section>
<section id="why-transcriptomics-6" class="slide level2 slide-only unlisted">
<h2>Why Transcriptomics?</h2>
<ul>
<li>Early techniques were often using large numbers of cells
<ul>
<li>Often multiple cell types within a biological sample</li>
</ul></li>
<li>Modern techniques are incredibly detailed
<ul>
<li>Single-Cell RNA characterises exact cell types and cell trajectories</li>
<li>Spatial transcriptomics used to identify co-located cells in tissue</li>
<li>Identify cell-cell signalling <em>in situ</em></li>
</ul></li>
</ul>
</section>
<section id="what-is-transcription" class="slide level2">
<h2>What Is Transcription</h2>
<div class="columns">
<div class="column" style="width:40%;">
<h3 id="definition-1">Definition</h3>
<blockquote>
<p>Transcription is the process of making an RNA copy of a gene sequence</p>
</blockquote>
</div><div class="column" style="width:60%;">
<div style="font-size: 90%; text-align: center;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="transcriptomics/transcription.jpg"></p>
<figcaption>Figure taken from <sup>1</sup> Licensed under CC-BY 4.0 by OpenStax</figcaption>
</figure>
</div>
</div>
</div></div>
<aside><ol class="aside-footnotes"><li id="fn1"><p>https://openoregon.pressbooks.pub/mhccbiology102/chapter/transcription/</p></li></ol></aside></section>
<section id="steps-of-transcription" class="slide level2">
<h2>Steps of Transcription</h2>
<ol type="1">
<li><strong>RNA polymerase binds to the promoter</strong> along with <span class="math inline">\(\geq1\)</span> transcription factors</li>
</ol>
<ol start="2" type="1">
<li class="fragment"><strong>RNA polymerase creates a transcription bubble</strong>
<ul>
<li class="fragment">separates the two DNA strands, breaking hydrogen bonds between complementary DNA nucleotides.</li>
</ul></li>
<li class="fragment"><strong>RNA polymerase adds RNA nucleotides</strong>
<ul>
<li class="fragment">complementary to the antisense DNA strand.</li>
</ul></li>
<li class="fragment"><strong>RNA sugar-phosphate backbone forms</strong></li>
<li class="fragment"><strong>Hydrogen bonds of the RNA–DNA complex break</strong> freeing the newly synthesized RNA strand.</li>
</ol>
</section>
<section id="steps-of-transcription-1" class="slide level2 slide-only unlisted">
<h2>Steps of Transcription</h2>
<h4 id="if-the-cell-is-a-eukaryotic-cell">If the cell is a eukaryotic cell</h4>
<ol start="6" type="1">
<li><strong>RNA processing</strong>
<ul>
<li>This may include polyadenylation, capping and splicing</li>
<li>Occurs during (or immediately after) transcription</li>
</ul></li>
<li><strong>RNA Localisation</strong>
<ul>
<li>The RNA may remain in the nucleus or exit to the cytoplasm through the nuclear pore complex</li>
</ul></li>
</ol>
<div class="fragment">
<ul>
<li>Eukaryotic mRNA, miRNA &amp; snRNA transcription uses <em>RNA Polymerase II</em>
<ul>
<li>RNA Pol I: rRNA</li>
<li>RNA Pol III: tRNA, 5S RNA some small RNAs</li>
</ul></li>
</ul>
</div>
</section>
<section id="eukaryotic-mrna-processing" class="slide level2">
<h2>Eukaryotic mRNA Processing</h2>
<ul>
<li>Nuclear mRNA have 5’ cap added
<ul>
<li>Protects single-stranded mRNA from degradation</li>
<li>Regulates nuclear export</li>
<li>Promotes translation</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>mRNAs are polyadenylated at the 3’ end
<ul>
<li>Also protects from degradation</li>
<li>Aids in transcription termination, export and translation</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Introns are spliced out as required<br>
</li>
</ul>
</div>
</section>
<section id="eukaryotic-mrna-processing-1" class="slide level2 slide-only unlisted">
<h2>Eukaryotic mRNA Processing</h2>

<img data-src="transcriptomics/eukaryotic_mrna_processing.svg" class="r-stretch quarto-figure-center"><p class="caption">Taken from <span class="citation" data-cites="Shafee2017-zv">Shafee and Lowe (<a href="#/references" role="doc-biblioref" onclick="">2017</a>)</span></p></section>
<section id="alternate-transcripts-and-isoforms" class="slide level2">
<h2>Alternate Transcripts and Isoforms</h2>

<img data-src="transcriptomics/DNA_alternative_splicing.png" class="r-stretch quarto-figure-center"><p class="caption">Image by the National Human Genome Research Institute</p></section>
<section id="transcriptome-resources" class="slide level2">
<h2>Transcriptome Resources</h2>
<ul>
<li>Reference Transcriptomes &amp; Genomes are now commonly available
<ul>
<li>Incorporate experimentally derived &amp; predicted sequences + loci</li>
</ul></li>
<li>Gencode<sup>1</sup> provide highest quality for mouse &amp; human
<ul>
<li>Release 48 (GRCh38): 78,686 genes + 385,669 transcripts</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Other organisms from Ensembl, RefSeq, UCSC etc
<ul>
<li>Zebrafish, Rat, Chicken, Drosophila, Wheat, Yeast, <em>E. Coli</em> etc</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Sometimes we build novel transcriptomes from specific tissues
<ul>
<li>e.g.&nbsp;sea snake venom gland, shiraz fruit</li>
</ul></li>
</ul>
</div>
<aside><ol class="aside-footnotes"><li id="fn2"><p>https://www.gencodegenes.org/</p></li></ol></aside></section></section>
<section>
<section id="early-transcriptomics" class="title-slide slide level1 center">
<h1>Early Transcriptomics</h1>

</section>
<section id="northern-blotting" class="slide level2">
<h2>Northern Blotting</h2>
<ul>
<li>Northern blot <span class="citation" data-cites="1977NorthernBlot">(<a href="#/references" role="doc-biblioref" onclick="">Alwine, Kemp, and Stark 1977</a>)</span> extended DNA-based methods (i.e Southern blot)
<ul>
<li>Earliest single-gene method</li>
</ul></li>
<li>Gel Electrophoresis then hybridisation with labelled probe
<ul>
<li>Requires some knowledge of RNA sequence</li>
</ul></li>
<li>Images scanned <span class="math inline">\(\rightarrow\)</span> <em>Densitometric Analysis</em> for crude quantitation</li>
<li>Possible for different isoforms to be detected
<ul>
<li>Sequence dependent</li>
</ul></li>
</ul>
</section>
<section id="rt-qpcr" class="slide level2">
<h2>RT-qPCR</h2>
<aside class="notes">
<p>The C<sub>T</sub> values as actually estimated to a decimal value</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li>“Gold-standard” for measurement of transcription levels
<ul>
<li>Single gene <span class="math inline">\(\implies\)</span> not a high-throughput technique</li>
</ul></li>
<li>Targets a single transcript region with specific primers to produce cDNA <br><span class="math inline">\(\rightarrow\)</span> Polymerase Chain Reaction (PCR)</li>
<li>Each PCR cycle approximately doubles the target region</li>
</ul>
<div class="fragment">
<ul>
<li>cDNA produced is identified using fluorophores
<ul>
<li>Fluorescence doubles with each cycle</li>
</ul></li>
<li>Once fluoresence passes a detection threshold, the cycle number is recorded
<ul>
<li>Known as the Cycle Threshold (C<sub>T</sub>) value</li>
</ul></li>
</ul>
</div>
</section>
<section id="rt-qpcr-1" class="slide level2 slide-only unlisted">
<h2>RT-qPCR</h2>

<img data-src="transcriptomics/qPCR.png" class="r-stretch quarto-figure-left"><p class="caption">A 10-fold dilution series</p></section>
<section id="rt-qpcr-2" class="slide level2 slide-only unlisted">
<h2>RT-qPCR</h2>
<ul>
<li>Higher C<sub>T</sub> values <span class="math inline">\(\implies\)</span> lower numbers of target molecule at the beginning</li>
<li>These can be used to estimate and compare abundance levels (i.e.&nbsp;gene expression)</li>
</ul>
<div class="fragment">
<ul>
<li>Is vulnerable to technical artefacts (e.g.&nbsp;pipetting variability)</li>
<li>Often includes one or more “housekeeper” genes thought to be stably expressed</li>
</ul>
</div>
<ul>
<li class="fragment">C<sub>T</sub> values are then <em>normalised</em> to the housekeeper genes <span class="math inline">\(\implies \Delta C_T\)</span>
<ul>
<li class="fragment">log<sub>2</sub> transformed values are used</li>
</ul></li>
<li class="fragment">Comparison between conditions is the change in <span class="math inline">\(\Delta C_T \implies \Delta\Delta C_T\)</span></li>
<li class="fragment">Represents change on the log<sub>2</sub> scale, i.e.&nbsp;<em>log fold-change</em></li>
</ul>
</section>
<section id="expressed-sequence-tags" class="slide level2">
<h2>Expressed Sequence Tags</h2>
<aside class="notes">
<ul>
<li>The senior author on the EST paper was J Craing Ventner who played an important role in the Human Genome Project</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li>The first attempt at capturing the larger transcriptome was ESTs <span class="citation" data-cites="1991VenterEST">(<a href="#/references" role="doc-biblioref" onclick="">Adams et al. 1991</a>)</span></li>
<li>Identified 609 human brain mRNA sequences
<ul>
<li>Selected for polyA-mRNA then reverse transcribed</li>
<li>Used random primers <span class="math inline">\(\rightarrow\)</span> Sanger Sequencing</li>
</ul></li>
<li>10 years before the Human Genome Project
<ul>
<li>Gene discovery was a hot topic</li>
</ul></li>
</ul>
</section>
<section id="sanger-sequencing" class="slide level2">
<h2>Sanger Sequencing</h2>

<img data-src="transcriptomics/Sanger-sequencing.svg" class="r-stretch quarto-figure-left"><p class="caption">Estevezj, <a href="https://creativecommons.org/licenses/by-sa/3.0">CC BY-SA 3.0</a>, via Wikimedia Commons</p></section>
<section id="sage-cage" class="slide level2">
<h2>SAGE &amp; CAGE</h2>
<ul>
<li>First high-throughput quantification method was <em>Serial Analysis of Gene Expression</em> (SAGE) <span class="citation" data-cites="pmid7570003">(<a href="#/references" role="doc-biblioref" onclick="">Velculescu et al. 1995</a>)</span></li>
</ul>
<div class="columns">
<div class="column" style="width:55%;">
<ul>
<li class="fragment">mRNA <span class="math inline">\(\rightarrow\)</span> cDNA using biotinylated primers</li>
<li class="fragment">cDNA bound to beads (using biotin) &amp; cleaved</li>
<li class="fragment">11mer “tags” were ligated into long sequenced using linker sequences</li>
<li class="fragment">Sequenced using Sanger Sequencing</li>
<li class="fragment">Deconvolution &amp; counting</li>
</ul>
</div><div class="column" style="width:45%;">
<div class="fragment">
<p><img data-src="transcriptomics/SAGE.png"></p>
</div>
</div></div>
</section>
<section id="sage-cage-1" class="slide level2 slide-only unlisted">
<h2>SAGE &amp; CAGE</h2>
<ul>
<li>The terminology of counting <em>tags</em> is still used by some manuals &amp; software to represent counting of sequences</li>
<li>Was described as Digital Gene Expression (DGE)
<ul>
<li>DGE still used but can be confused with Differential Gene Expression</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>A variant called Cap Analysis of Gene Expression (CAGE) targeted the 5’ Cap</li>
<li>Heavily used by FANTOM project <span class="citation" data-cites="Abugessaisa2021-fn">(<a href="#/references" role="doc-biblioref" onclick="">Abugessaisa et al. 2020</a>)</span> to identify exact Transcription Start Sites (TSS)</li>
</ul>
</div>
</section>
<section id="microarray-technology" class="slide level2">
<h2>Microarray Technology</h2>
</section>
<section id="references" class="slide level2 smaller scrollable">
<h2>References</h2>

<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-Abugessaisa2021-fn" class="csl-entry" role="listitem">
Abugessaisa, Imad, Jordan A Ramilowski, Marina Lizio, Jesicca Severin, Akira Hasegawa, Jayson Harshbarger, Atsushi Kondo, et al. 2020. <span>“FANTOM Enters 20th Year: Expansion of Transcriptomic Atlases and Functional Annotation of Non-Coding RNAs.”</span> <em>Nucleic Acids Research</em> 49 (D1): D892–98. <a href="https://doi.org/10.1093/nar/gkaa1054">https://doi.org/10.1093/nar/gkaa1054</a>.
</div>
<div id="ref-1991VenterEST" class="csl-entry" role="listitem">
Adams, Mark D., Jenny M. Kelley, Jeannine D. Gocayne, Mark Dubnick, Mihael H. Polymeropoulos, Hong Xiao, Carl R. Merril, et al. 1991. <span>“Complementary DNA Sequencing: Expressed Sequence Tags and Human Genome Project.”</span> <em>Science</em> 252 (5013): 1651–56. <a href="http://www.jstor.org/stable/2876333">http://www.jstor.org/stable/2876333</a>.
</div>
<div id="ref-1977NorthernBlot" class="csl-entry" role="listitem">
Alwine, J. C., D. J. Kemp, and G. R. Stark. 1977. <span>“<span class="nocase"><span>M</span>ethod for detection of specific <span>R</span><span>N</span><span>A</span>s in agarose gels by transfer to diazobenzyloxymethyl-paper and hybridization with <span>D</span><span>N</span><span>A</span> probes</span>.”</span> <em>Proc. Natl. Acad. Sci. U.S.A.</em> 74 (12): 5350–54.
</div>
<div id="ref-Fang2015-uq" class="csl-entry" role="listitem">
Fang, Rui, Walter N Moss, Michael Rutenberg-Schoenberg, and Matthew D Simon. 2015. <span>“Probing Xist <span>RNA</span> Structure in Cells Using Targeted Structure-Seq.”</span> <em>PLoS Genet.</em> 11 (12): e1005668.
</div>
<div id="ref-Shafee2017-zv" class="csl-entry" role="listitem">
Shafee, Thomas, and Rohan Lowe. 2017. <span>“Eukaryotic and Prokaryotic Gene Structure.”</span> <em><span>WikiJournal</span> of Medicine</em>, January. <a href="https://doi.org/10.15347/WJM/2017.002">https://doi.org/10.15347/WJM/2017.002</a>.
</div>
<div id="ref-pmid7570003" class="csl-entry" role="listitem">
Velculescu, V. E., L. Zhang, B. Vogelstein, and K. W. Kinzler. 1995. <span>“<span class="nocase"><span>S</span>erial analysis of gene expression</span>.”</span> <em>Science</em> 270 (5235): 484–87.
</div>
<div id="ref-Wang2009-hf" class="csl-entry" role="listitem">
Wang, Zhong, Mark Gerstein, and Michael Snyder. 2009. <span>“<span>RNA</span>-Seq: A Revolutionary Tool for Transcriptomics.”</span> <em>Nat. Rev. Genet.</em> 10 (1): 57–63.
</div>
</div>
</section></section>

    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">

</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1280,

        height: 720,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <div id="logo-tki" style="position: fixed; z-index: 1000;">
      <img src="https://www.thekids.org.au/Static/Theme/images/TheKids-RGB_Logo_Colour-Pattern.svg" style="height: 80px;">
    </div>
    <div id="logo-bodl" style="position: fixed; z-index: 1000;">
      <img src="https://raw.githubusercontent.com/black-ochre-data-labs/20250902_ASI_RWorkshop/refs/heads/main/assets/bodl_logo_white_background.jpg" style="height: 80px;">
    </div>
    <div id="logo-uoa" style="z-index: 1000;">
      <img src="transcriptomics/UoA_logo_col_vert.png" style="height: 80px;">
    </div>


    <script>
    function positionLogos() {
      const tkiLogo = document.getElementById('logo-tki');
      const uoaLogo = document.getElementById('logo-uoa');
      const bodlLogo = document.getElementById('logo-bodl');

      // Check if this is RevealJS
      const isRevealJS = document.body.classList.contains('reveal') ||
                         document.querySelector('.reveal') !== null ||
                         typeof Reveal !== 'undefined';

      if (isRevealJS) {
        // RevealJS - both logos at bottom
        tkiLogo.style.bottom = '10px';
        tkiLogo.style.right = '340px';
        tkiLogo.style.top = 'auto';
        tkiLogo.style.display = 'block';

        bodlLogo.style.bottom = '10px';
        bodlLogo.style.right = '160px';
        bodlLogo.style.top = 'auto';
        bodlLogo.style.display = 'block';

        uoaLogo.style.bottom = '10px';
        uoaLogo.style.right = '40px';
        uoaLogo.style.top = 'auto';
        uoaLogo.style.display = 'block';
        uoaLogo.style.position = 'fixed';
      } else {
        // Flat HTML - only right logo at top
        tkiLogo.style.display = 'none';
        bodlLogo.style.display = 'none';

        uoaLogo.style.top = '30px';
        uoaLogo.style.left = '60%';
        uoaLogo.style.bottom = 'auto';
        uoaLogo.style.position = 'absolute';
        uoaLogo.style.display = 'block';
      }
    }

    // Run positioning
    document.addEventListener('DOMContentLoaded', positionLogos);
    setTimeout(positionLogos, 100);
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
            const codeEl = trigger.previousElementSibling.cloneNode(true);
            for (const childEl of codeEl.children) {
              if (isCodeAnnotation(childEl)) {
                childEl.remove();
              }
            }
            return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp('/' + window.location.host + '/');
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>