<!DOCTYPE html>
<html lang="en"><head>
<script src="transcriptomics-lecture1-slides_files/libs/clipboard/clipboard.min.js"></script>
<script src="transcriptomics-lecture1-slides_files/libs/quarto-html/tabby.min.js"></script>
<script src="transcriptomics-lecture1-slides_files/libs/quarto-html/popper.min.js"></script>
<script src="transcriptomics-lecture1-slides_files/libs/quarto-html/tippy.umd.min.js"></script>
<link href="transcriptomics-lecture1-slides_files/libs/quarto-html/tippy.css" rel="stylesheet">
<link href="transcriptomics-lecture1-slides_files/libs/quarto-html/light-border.css" rel="stylesheet">
<link href="transcriptomics-lecture1-slides_files/libs/quarto-html/quarto-syntax-highlighting-37eea08aefeeee20ff55810ff984fec1.css" rel="stylesheet" id="quarto-text-highlighting-styles"><meta charset="utf-8">
  <meta name="generator" content="quarto-1.7.32">

  <meta name="author" content="Dr Stevie Pederson (They/Them) stevie.pederson@thekids.org.au">
  <meta name="dcterms.date" content="2025-09-01">
  <title>Transcriptomics: Lecture 1</title>
  <meta name="apple-mobile-web-app-capable" content="yes">
  <meta name="apple-mobile-web-app-status-bar-style" content="black-translucent">
  <meta name="viewport" content="width=device-width, initial-scale=1.0, maximum-scale=1.0, user-scalable=no, minimal-ui">
  <link rel="stylesheet" href="transcriptomics-lecture1-slides_files/libs/revealjs/dist/reset.css">
  <link rel="stylesheet" href="transcriptomics-lecture1-slides_files/libs/revealjs/dist/reveal.css">
  <style>
    code{white-space: pre-wrap;}
    span.smallcaps{font-variant: small-caps;}
    div.columns{display: flex; gap: min(4vw, 1.5em);}
    div.column{flex: auto; overflow-x: auto;}
    div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
    ul.task-list{list-style: none;}
    ul.task-list li input[type="checkbox"] {
      width: 0.8em;
      margin: 0 0.8em 0.2em -1em; /* quarto-specific, see https://github.com/quarto-dev/quarto-cli/issues/4556 */ 
      vertical-align: middle;
    }
    /* CSS for citations */
    div.csl-bib-body { }
    div.csl-entry {
      clear: both;
      margin-bottom: 0em;
    }
    .hanging-indent div.csl-entry {
      margin-left:2em;
      text-indent:-2em;
    }
    div.csl-left-margin {
      min-width:2em;
      float:left;
    }
    div.csl-right-inline {
      margin-left:2em;
      padding-left:1em;
    }
    div.csl-indent {
      margin-left: 2em;
    }  </style>
  <link rel="stylesheet" href="transcriptomics-lecture1-slides_files/libs/revealjs/dist/theme/quarto-6a216ba96c5c25e91d68b44b6d1a2940.css">
  <link href="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.css" rel="stylesheet">
  <link href="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/reveal-menu/menu.css" rel="stylesheet">
  <link href="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/reveal-menu/quarto-menu.css" rel="stylesheet">
  <link href="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/quarto-support/footer.css" rel="stylesheet">
  <style type="text/css">
    .reveal div.sourceCode {
      margin: 0;
      overflow: auto;
    }
    .reveal div.hanging-indent {
      margin-left: 1em;
      text-indent: -1em;
    }
    .reveal .slide:not(.center) {
      height: 100%;
    }
    .reveal .slide.scrollable {
      overflow-y: auto;
    }
    .reveal .footnotes {
      height: 100%;
      overflow-y: auto;
    }
    .reveal .slide .absolute {
      position: absolute;
      display: block;
    }
    .reveal .footnotes ol {
      counter-reset: ol;
      list-style-type: none; 
      margin-left: 0;
    }
    .reveal .footnotes ol li:before {
      counter-increment: ol;
      content: counter(ol) ". "; 
    }
    .reveal .footnotes ol li > p:first-child {
      display: inline-block;
    }
    .reveal .slide ul,
    .reveal .slide ol {
      margin-bottom: 0.5em;
    }
    .reveal .slide ul li,
    .reveal .slide ol li {
      margin-top: 0.4em;
      margin-bottom: 0.2em;
    }
    .reveal .slide ul[role="tablist"] li {
      margin-bottom: 0;
    }
    .reveal .slide ul li > *:first-child,
    .reveal .slide ol li > *:first-child {
      margin-block-start: 0;
    }
    .reveal .slide ul li > *:last-child,
    .reveal .slide ol li > *:last-child {
      margin-block-end: 0;
    }
    .reveal .slide .columns:nth-child(3) {
      margin-block-start: 0.8em;
    }
    .reveal blockquote {
      box-shadow: none;
    }
    .reveal .tippy-content>* {
      margin-top: 0.2em;
      margin-bottom: 0.7em;
    }
    .reveal .tippy-content>*:last-child {
      margin-bottom: 0.2em;
    }
    .reveal .slide > img.stretch.quarto-figure-center,
    .reveal .slide > img.r-stretch.quarto-figure-center {
      display: block;
      margin-left: auto;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-left,
    .reveal .slide > img.r-stretch.quarto-figure-left  {
      display: block;
      margin-left: 0;
      margin-right: auto; 
    }
    .reveal .slide > img.stretch.quarto-figure-right,
    .reveal .slide > img.r-stretch.quarto-figure-right  {
      display: block;
      margin-left: auto;
      margin-right: 0; 
    }
  </style>
</head>
<body class="quarto-light">
  <div class="reveal">
    <div class="slides">

<section id="title-slide" class="quarto-title-block center">
  <h1 class="title">Transcriptomics: Lecture 1</h1>
  <p class="subtitle"></p><div class="line-block">Biotech 7005/Bioinf 3000<br>
Frontiers of Biotechnology: Bioinformatics and Systems Modelling<br>
The University of Adelaide</div><p></p>

<div class="quarto-title-authors">
<div class="quarto-title-author">
<div class="quarto-title-author-name">
Dr Stevie Pederson (They/Them)<br>stevie.pederson@thekids.org.au 
</div>
        <p class="quarto-title-affiliation">
            </p><div class="line-block">Black Ochre Data Labs<br>
            The Kids Research Institute Australia</div>
          <p></p>
    </div>
</div>

  <p class="date">September 1, 2025</p>
</section>
<section id="helpful-links" class="slide level2">
<h2>Helpful Links</h2>
<ul>
<li>The slides are available <a href="./transcriptomics-lecture1-slides.html">here</a></li>
<li>The same notes are available as a single-page html <a href="./transcriptomics1-lecture1.html">here</a></li>
</ul>
</section>
<section id="welcome-to-country" class="slide level2">
<h2><span class="text-red">Welcome To Country</span></h2>
<div class="text-red">
<p><em>I’d like to acknowledge the Kaurna people as the traditional owners and custodians of the land we know today as the Adelaide Plains, where I live &amp; work.</em> <br><br> <em>I also acknowledge the deep feelings of attachment and relationship of the Kaurna people to their place.</em> <br><br> <em>I pay my respects to the cultural authority of Aboriginal and Torres Strait Islander peoples from other areas of Australia, and pay our respects to Elders past, present and emerging, and acknowledge any Aboriginal Australians who may be with us today</em></p>
</div>
</section>
<section>
<section id="introduction-to-transcriptomics" class="title-slide slide level1 center">
<h1>Introduction To Transcriptomics</h1>

</section>
<section id="introduction" class="slide level2">
<h2>Introduction</h2>
<ul>
<li>Postdoctoral Bioinformatician, Black Ochre Data Labs, Adelaide</li>
<li>Working in collaboration with members of the SA Aboriginal community</li>
<li>Multi-omics project to identify and address the underlying causes of high T2D rates and complications
<ul>
<li>Using genomics, epigenomics, transcriptomics and other layers</li>
<li>My focus is on the transcriptomics layer</li>
</ul></li>
</ul>
</section>
<section id="why-transcriptomics" class="slide level2">
<h2>Why Transcriptomics?</h2>
<div class="columns">
<div class="column">
<ul>
<li>DNA can be described as being like a giant book of instructions</li>
<li>Some regions are defined as <em>genes</em>
<ul>
<li>Originally considered to be the basic unit of inheritance</li>
<li>Now commonly used to describe a region of DNA transcribed into RNA</li>
</ul></li>
</ul>
</div><div class="column">
<div class="fragment">
<div style="font-size: 60%; text-align: center;">
<p><img data-src="transcriptomics/DNA_to_protein_or_ncRNA.png"> <br>By <a href="//commons.wikimedia.org/wiki/User:Evolution_and_evolvability" title="User:Evolution and evolvability">Thomas Shafee</a> - <span class="int-own-work" lang="en">Own work</span>, <a href="https://creativecommons.org/licenses/by/4.0" title="Creative Commons Attribution 4.0">CC BY 4.0</a>, <a href="https://commons.wikimedia.org/w/index.php?curid=39441809">Wikimedia Link</a></p>
</div>
</div>
</div></div>
</section>
<section id="why-transcriptomics-1" class="slide level2 slide-only unlisted">
<h2>Why Transcriptomics?</h2>
<ul>
<li>DNA <span class="math inline">\(\rightarrow\)</span> mRNA <span class="math inline">\(\rightarrow\)</span> Proteins
<ul>
<li>Commonly referred to as the Central Dogma of Biology</li>
</ul></li>
<li>Proteins are the <em>workhorses</em> of the cell &amp; body
<ul>
<li>Do most of the work, and are responsible for most of the structure</li>
<li>Examples like keratin (hair), haemoglobin (oxygen transport) etc</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>ncRNAs are also highly functional
<ul>
<li>Ribosomal RNA (rRNA) + tRNA <span class="math inline">\(\rightarrow\)</span> translation from mRNA to Protein</li>
<li>microRNAs play a role in gene-regulation via mRNA stability</li>
</ul></li>
</ul>
</div>
</section>
<section id="why-transcriptomics-2" class="slide level2 slide-only unlisted">
<h2>Why Transcriptomics?</h2>
<div class="columns">
<div class="column">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="transcriptomics/pgen.1005668.g006.png"></p>
<figcaption>Taken from <span class="citation" data-cites="Fang2015-uq">Fang et al. (<a href="#/references" role="doc-biblioref" onclick="">2015</a>)</span></figcaption>
</figure>
</div>
</div><div class="column">
<ul>
<li>Most RNA is single-stranded but can have extremely complex structure
<ul>
<li>Shown is a 2kb region from the lncRNA Xist (17kb in total)</li>
</ul></li>
<li>Coats the <em>entire X chromosome</em> during X inactivation</li>
<li>Also interacts with the antisense lncRNA Tsix</li>
</ul>
</div></div>
</section>
<section id="why-transcriptomics-3" class="slide level2 slide-only unlisted">
<h2>Why Transcriptomics?</h2>
<h3 id="definition">Definition</h3>
<p>Based on <span class="citation" data-cites="Wang2009-hf">Wang, Gerstein, and Snyder (<a href="#/references" role="doc-biblioref" onclick="">2009</a>)</span></p>
<blockquote>
<p>The transcriptome can be defined as the complete set of (RNA) transcripts in a cell, or a population of cells, for a specific developmental stage or physiological condition</p>
</blockquote>
<div class="fragment">
<ul>
<li>Transcriptomics is simply the study of the transcriptome</li>
<li>Can be the <em>entire RNA content of a cell</em> (or cells) or a subset of molecules (e.g.&nbsp;mRNA, miRNA)</li>
</ul>
</div>
</section>
<section id="why-transcriptomics-4" class="slide level2 slide-only unlisted">
<h2>Why Transcriptomics?</h2>
<ul>
<li>Is a snapshot of the <strong>dynamic biological processes</strong> associated with a biological question</li>
<li>Use to make inference about these processes
<ul>
<li>Identify therapeutic targets for Cardiovascular Disease</li>
<li>Biomarkers for CAR-T cells</li>
<li>Key drivers of correlated gene networks</li>
<li>Early drivers of neurodegeneration in Alzheimer’s</li>
</ul></li>
<li>Assumed to be low-level
<ul>
<li>DNA <span class="math inline">\(\rightarrow\)</span> <span class="text-red">RNA</span> <span class="fragment"><span class="math inline">\(\rightarrow\)</span> Protein <span class="math inline">\(\rightarrow\)</span> Metabolites, Signalling molecules, etc …</span></li>
</ul></li>
</ul>
</section>
<section id="why-transcriptomics-5" class="slide level2 slide-only unlisted">
<h2>Why Transcriptomics?</h2>
<ul>
<li>Is the lowest molecular level where quantity becomes a key aspect
<ul>
<li>Highly-expressed, or low-expressed genes are important</li>
<li>Changes in response to stimulus impact gene expression levels</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Much of the early transcriptomic analyses were quantitative
<ul>
<li>Sequence variation often captured at DNA-level</li>
</ul></li>
<li>Now extending to transcript structure and modifications
<ul>
<li>Identification of fusion transcripts, RNA-methylation etc</li>
</ul></li>
</ul>
</div>
</section>
<section id="why-transcriptomics-6" class="slide level2 slide-only unlisted">
<h2>Why Transcriptomics?</h2>
<ul>
<li>Early techniques were often using large numbers of cells
<ul>
<li>Often multiple cell types within a biological sample</li>
</ul></li>
<li>Modern techniques are incredibly detailed
<ul>
<li>Single-Cell RNA characterises exact cell types and cell trajectories</li>
<li>Spatial transcriptomics used to identify co-located cells in tissue</li>
<li>Identify cell-cell signalling <em>in situ</em></li>
</ul></li>
</ul>
</section>
<section id="what-is-transcription" class="slide level2">
<h2>What Is Transcription</h2>
<div class="columns">
<div class="column" style="width:40%;">
<h3 id="definition-1">Definition</h3>
<blockquote>
<p>Transcription is the process of making an RNA copy of a gene sequence</p>
</blockquote>
</div><div class="column" style="width:60%;">
<div style="font-size: 90%; text-align: center;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="transcriptomics/transcription.jpg"></p>
<figcaption>Figure taken from <sup>1</sup> Licensed under CC-BY 4.0 by OpenStax</figcaption>
</figure>
</div>
</div>
</div></div>
<aside><ol class="aside-footnotes"><li id="fn1"><p>https://openoregon.pressbooks.pub/mhccbiology102/chapter/transcription/</p></li></ol></aside></section>
<section id="steps-of-transcription" class="slide level2">
<h2>Steps of Transcription</h2>
<ol type="1">
<li><strong>RNA polymerase binds to the promoter</strong> along with <span class="math inline">\(\geq1\)</span> transcription factors</li>
</ol>
<ol start="2" type="1">
<li class="fragment"><strong>RNA polymerase creates a transcription bubble</strong>
<ul>
<li class="fragment">separates the two DNA strands, breaking hydrogen bonds between complementary DNA nucleotides.</li>
</ul></li>
<li class="fragment"><strong>RNA polymerase adds RNA nucleotides</strong>
<ul>
<li class="fragment">complementary to the antisense DNA strand.</li>
</ul></li>
<li class="fragment"><strong>RNA sugar-phosphate backbone forms</strong></li>
<li class="fragment"><strong>Hydrogen bonds of the RNA–DNA complex break</strong> freeing the newly synthesized RNA strand.</li>
</ol>
</section>
<section id="steps-of-transcription-1" class="slide level2 slide-only unlisted">
<h2>Steps of Transcription</h2>
<h4 id="if-the-cell-is-a-eukaryotic-cell">If the cell is a eukaryotic cell</h4>
<ol start="6" type="1">
<li><strong>RNA processing</strong>
<ul>
<li>This may include polyadenylation, capping and splicing</li>
<li>Occurs during (or immediately after) transcription</li>
</ul></li>
<li><strong>RNA Localisation</strong>
<ul>
<li>The RNA may remain in the nucleus or exit to the cytoplasm through the nuclear pore complex</li>
</ul></li>
</ol>
<div class="fragment">
<ul>
<li>Eukaryotic mRNA, miRNA &amp; snRNA transcription uses <em>RNA Polymerase II</em>
<ul>
<li>RNA Pol I: rRNA</li>
<li>RNA Pol III: tRNA, 5S RNA some small RNAs</li>
</ul></li>
</ul>
</div>
</section>
<section id="eukaryotic-mrna-processing" class="slide level2">
<h2>Eukaryotic mRNA Processing</h2>
<ul>
<li>Nuclear mRNA have 5’ cap added
<ul>
<li>Protects single-stranded mRNA from degradation</li>
<li>Regulates nuclear export</li>
<li>Promotes translation</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>mRNAs are polyadenylated at the 3’ end
<ul>
<li>Also protects from degradation</li>
<li>Aids in transcription termination, export and translation</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Introns are spliced out as required<br>
</li>
</ul>
</div>
</section>
<section id="eukaryotic-mrna-processing-1" class="slide level2 slide-only unlisted">
<h2>Eukaryotic mRNA Processing</h2>

<img data-src="transcriptomics/eukaryotic_mrna_processing.svg" class="r-stretch quarto-figure-center"><p class="caption">Taken from <span class="citation" data-cites="Shafee2017-zv">Shafee and Lowe (<a href="#/references" role="doc-biblioref" onclick="">2017</a>)</span></p></section>
<section id="alternate-transcripts-and-isoforms" class="slide level2">
<h2>Alternate Transcripts and Isoforms</h2>

<img data-src="transcriptomics/DNA_alternative_splicing.png" class="r-stretch quarto-figure-center"><p class="caption">Image by the National Human Genome Research Institute</p></section>
<section id="transcriptome-resources" class="slide level2">
<h2>Transcriptome Resources</h2>
<ul>
<li>Reference Transcriptomes &amp; Genomes are now commonly available
<ul>
<li>Incorporate experimentally derived &amp; predicted sequences + loci</li>
</ul></li>
<li>Gencode<sup>1</sup> provide highest quality for mouse &amp; human
<ul>
<li>Release 48 (GRCh38): 78,686 genes + 385,669 transcripts</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>Other organisms from Ensembl, RefSeq, UCSC etc
<ul>
<li>Zebrafish, Rat, Chicken, Drosophila, Wheat, Yeast, <em>E. Coli</em> etc</li>
</ul></li>
</ul>
</div>
<div class="fragment">
<ul>
<li>Sometimes we build novel transcriptomes from specific tissues
<ul>
<li>e.g.&nbsp;sea snake venom gland, shiraz fruit</li>
</ul></li>
</ul>
</div>
<aside><ol class="aside-footnotes"><li id="fn2"><p>https://www.gencodegenes.org/</p></li></ol></aside></section></section>
<section>
<section id="early-transcriptomics" class="title-slide slide level1 center">
<h1>Early Transcriptomics</h1>

</section>
<section id="northern-blotting" class="slide level2">
<h2>Northern Blotting</h2>
<ul>
<li>Northern blot <span class="citation" data-cites="1977NorthernBlot">(<a href="#/references" role="doc-biblioref" onclick="">Alwine, Kemp, and Stark 1977</a>)</span> extended DNA-based methods (i.e Southern blot) <span class="fragment"><span class="math inline">\(\implies\)</span> Earliest single-gene method</span></li>
</ul>
<div class="fragment">
<ul>
<li>Gel Electrophoresis then hybridisation with labelled probe
<ul>
<li>Requires some knowledge of RNA sequence</li>
</ul></li>
<li>Images scanned <span class="math inline">\(\rightarrow\)</span> <em>Densitometric Analysis</em> for crude quantitation</li>
<li>Possible for different isoforms to be detected
<ul>
<li>Sequence dependent</li>
</ul></li>
</ul>
</div>
</section>
<section id="rt-qpcr" class="slide level2">
<h2>RT-qPCR</h2>
<aside class="notes">
<p>The C<sub>T</sub> values is actually estimated to a decimal value</p>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li>“Gold-standard” for measurement of transcription levels
<ul>
<li>Single gene <span class="math inline">\(\implies\)</span> not a high-throughput technique</li>
</ul></li>
<li>Targets a single transcript region with specific primers to produce cDNA <br><span class="math inline">\(\rightarrow\)</span> Polymerase Chain Reaction (PCR)</li>
<li>Each PCR cycle approximately doubles the target region</li>
</ul>
<div class="fragment">
<ul>
<li>cDNA produced is identified using fluorophores
<ul>
<li>Fluorescence doubles with each cycle</li>
</ul></li>
<li>Once fluorescence passes a detection threshold, the cycle number is recorded
<ul>
<li>Known as the Cycle Threshold (C<sub>T</sub>) value</li>
</ul></li>
</ul>
</div>
</section>
<section id="rt-qpcr-1" class="slide level2 slide-only unlisted">
<h2>RT-qPCR</h2>

<img data-src="transcriptomics/qPCR.png" class="r-stretch quarto-figure-left"><p class="caption">A 10-fold dilution series</p></section>
<section id="rt-qpcr-2" class="slide level2 slide-only unlisted">
<h2>RT-qPCR</h2>
<ul>
<li>Higher C<sub>T</sub> values <span class="math inline">\(\implies\)</span> lower numbers of target molecule at the beginning</li>
<li>These can be used to estimate and compare abundance levels (i.e.&nbsp;gene expression)</li>
</ul>
<div class="fragment">
<ul>
<li>Is vulnerable to technical artefacts (e.g.&nbsp;pipetting variability)</li>
<li>Often includes one or more “housekeeper” genes thought to be stably expressed</li>
</ul>
</div>
<ul>
<li class="fragment">C<sub>T</sub> values are <em>normalised</em> to the housekeeper genes <span class="math inline">\(\implies \Delta C_T\)</span>
<ul>
<li class="fragment">log<sub>2</sub> transformed values are used: <span class="math inline">\(\Delta C_{T_g} = \log_2 C_{T_g} - \log_2 C_{T_{hk}}\)</span></li>
</ul></li>
<li class="fragment">Comparison between conditions is the change in <span class="math inline">\(\Delta C_T \implies \Delta\Delta C_T\)</span></li>
<li class="fragment">Represents change on the log<sub>2</sub> scale, i.e.&nbsp;<em>log fold-change</em></li>
</ul>
</section>
<section id="expressed-sequence-tags" class="slide level2">
<h2>Expressed Sequence Tags</h2>
<aside class="notes">
<ul>
<li>The senior author on the EST paper was J Craig Ventner who played an important role in the Human Genome Project</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li>The first attempt at capturing the larger transcriptome was ESTs <span class="citation" data-cites="1991VenterEST">(<a href="#/references" role="doc-biblioref" onclick="">Adams et al. 1991</a>)</span></li>
<li>Identified 609 human brain mRNA sequences
<ul>
<li>Selected for polyA-mRNA then reverse transcribed</li>
<li>Used random primers <span class="math inline">\(\rightarrow\)</span> Sanger Sequencing</li>
</ul></li>
<li>10 years before the Human Genome Project
<ul>
<li>Gene discovery was a hot topic</li>
</ul></li>
</ul>
</section>
<section id="sanger-sequencing" class="slide level2">
<h2>Sanger Sequencing</h2>

<img data-src="transcriptomics/Sanger-sequencing.svg" class="r-stretch quarto-figure-left"><p class="caption">Estevezj, <a href="https://creativecommons.org/licenses/by-sa/3.0">CC BY-SA 3.0</a>, via Wikimedia Commons</p></section>
<section id="sage-cage" class="slide level2">
<h2>SAGE &amp; CAGE</h2>
<ul>
<li>First high-throughput quantification method was <em>Serial Analysis of Gene Expression</em> (SAGE) <span class="citation" data-cites="pmid7570003">(<a href="#/references" role="doc-biblioref" onclick="">Velculescu et al. 1995</a>)</span></li>
</ul>
<div class="columns">
<div class="column" style="width:58%;">
<ul>
<li class="fragment">mRNA <span class="math inline">\(\rightarrow\)</span> cDNA using biotinylated primers</li>
<li class="fragment">cDNA bound to beads (using biotin) &amp; cleaved</li>
<li class="fragment">11mer “tags” were ligated into long sequenced using linker sequences</li>
<li class="fragment">Sequenced using Sanger Sequencing</li>
<li class="fragment">Deconvolution &amp; counting</li>
</ul>
</div><div class="column" style="width:42%;">
<div style="font-size: 80%; text-align: center;">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="transcriptomics/SAGE.png"></p>
<figcaption>Thomas Shafee, <a href="https://creativecommons.org/licenses/by/4.0">CC BY 4.0</a>, via Wikimedia Commons</figcaption>
</figure>
</div>
</div>
</div></div>
</section>
<section id="sage-cage-1" class="slide level2 slide-only unlisted">
<h2>SAGE &amp; CAGE</h2>
<ul>
<li>The terminology of counting <em>tags</em> is still used by some manuals &amp; software
<ul>
<li>Statistical models still form the basis of modern transcriptomics</li>
</ul></li>
<li>Was described as Digital Gene Expression (DGE)
<ul>
<li>The term DGE is still used but easily confused with Differential Gene Expression</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>A variant called Cap Analysis of Gene Expression (CAGE) targeted the 5’ Cap</li>
<li>Heavily used by FANTOM project <span class="citation" data-cites="Abugessaisa2021-fn">(<a href="#/references" role="doc-biblioref" onclick="">Abugessaisa et al. 2020</a>)</span> to identify exact Transcription Start Sites (TSS)</li>
</ul>
</div>
</section>
<section id="microarray-technology" class="slide level2">
<h2>Microarray Technology</h2>
<aside class="notes">
<ul>
<li>My search last week showed 69000 public microarray datasets in the GEO database</li>
<li>I reviewed a Scientific Reports submission using public array data last month</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li>Microarrays represent the <em>birth of modern transcriptomics</em>
<ul>
<li>Thousands of genes could be measured simultaneously!!!</li>
<li>Tens of thousands of public datasets <span class="math inline">\(\implies\)</span> still being mined</li>
</ul></li>
<li>Established concurrently with the Human Genome Project (1990-2003)
<ul>
<li>Databases &amp; complete reference sequences become widely available</li>
</ul></li>
</ul>
<ul>
<li class="fragment">All require fluorescently labelled cDNA copies of RNA</li>
<li class="fragment">Hybridised to the array using probes for known sequences
<ul>
<li class="fragment"><span class="math inline">\(\uparrow\)</span> fluorescence <span class="math inline">\(\implies \uparrow\)</span> RNA abundance</li>
</ul></li>
</ul>
</section>
<section id="microarray-technology-1" class="slide level2 slide-only unlisted">
<h2>Microarray Technology</h2>
<ul>
<li>All microarrays follow the same basic process</li>
</ul>

<img data-src="transcriptomics/Microarray_exp_horizontal.svg" class="r-stretch quarto-figure-center"><p class="caption">Image courtesy of Squidonius, Public domain, via Wikimedia Commons</p></section>
<section id="two-colour-arrays" class="slide level2">
<h2>Two Colour Arrays</h2>
<div class="columns">
<div class="column" style="width:57%;">
<ul>
<li>Two colour microarrays were printed microscope slides</li>
<li>Known probe sequences were <em>printed</em> to the surface in defined locations
<ul>
<li>60-75mer oligonucleotide probes</li>
<li>Highly customisable by project</li>
<li>Two samples per array</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>One sample labelled with Cy5 (Red)</li>
<li>Other sample labelled with Cy3 (Green)</li>
<li>Scanned at 570nm (Cy3) and 670nm (Cy5)</li>
</ul>
</div>
</div><div class="column" style="width:43%;">
<div class="fragment">
<div class="quarto-figure quarto-figure-left">
<figure>
<p><img data-src="transcriptomics/2colour.jpg" style="width:100.0%"></p>
<figcaption>Section of two-colour array taken from <span class="citation" data-cites="Shalon01071996">Shalon, Smith, and Brown (<a href="#/references" role="doc-biblioref" onclick="">1996</a>)</span></figcaption>
</figure>
</div>
</div>
</div></div>
</section>
<section id="ma-plots" class="slide level2">
<h2>MA Plots</h2>
<div class="columns">
<div class="column">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="transcriptomics/dudoit_ma_plot.png"></p>
<figcaption>Image from <span class="citation" data-cites="dudoit02-ma">Dudoit et al. (<a href="#/references" role="doc-biblioref" onclick="">2002</a>)</span></figcaption>
</figure>
</div>
</div><div class="column">
<ul>
<li>Mean of Differences<br><span class="math inline">\(M = \log_2(\frac{R}{G}) = \log_2(R) - \log_2(G)\)</span></li>
<li>Average Signal<br><span class="math inline">\(A = \frac{1}{2}\log_2(RG) = \frac{\log_2(R) + \log_2(G)}{2}\)</span></li>
</ul>
<div class="fragment">
<ul>
<li>Used to assess any bias</li>
<li>Also to show DE genes</li>
</ul>
</div>
</div></div>
</section>
<section id="single-channel-arrays" class="slide level2">
<h2>Single Channel Arrays</h2>
<div class="columns">
<div class="column" style="width:55%;">
<ul>
<li>Affymetrix Arrays became dominant
<ul>
<li>Factory manufactured</li>
</ul></li>
<li>Standardised layout for each organism</li>
<li>Single sample per array
<ul>
<li>Only scanned at one frequency</li>
</ul></li>
<li>More genes/array</li>
</ul>
<div class="fragment">
<ul>
<li>25mer probes targeting 3’ end of transcript
<ul>
<li>Captured only intact transcripts</li>
</ul></li>
</ul>
</div>
</div><div class="column" style="width:45%;">
<div style="font-size: 80%">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="transcriptomics/Affymetrix.jpg"></p>
<figcaption>No author known. Schutz assumed based on copyright claims. <a href="http://creativecommons.org/licenses/by-sa/3.0/">CC BY-SA 3.0</a>, via Wikimedia Commons</figcaption>
</figure>
</div>
</div>
</div></div>
</section>
<section id="single-channel-arrays-1" class="slide level2 slide-only unlisted">
<h2>Single Channel Arrays</h2>
<div class="columns">
<div class="column" style="width:85%;">
<p><br></p>
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="transcriptomics/microarrayLayout.jpg" style="width:100.0%"></p>
<figcaption>The basic methodology underpinning Affymetrix array design. Source unknown</figcaption>
</figure>
</div>
</div></div>
</section>
<section id="arrays" class="slide level2">
<h2>3’ Arrays</h2>
<ul>
<li>Each 3’ exon targeted by 11 unique 25mer probes <span class="math inline">\(\implies\)</span> <em>a probeset</em></li>
<li>Possible to detect different transcripts only if 3’ exons differ</li>
</ul>
<ul>
<li class="fragment"><em>Perfect Match</em> (PM) probes <span class="math inline">\(\implies\)</span> exactly matches target sequence
<ul>
<li class="fragment">Known to capture off-target signal <span class="math inline">\(\implies\)</span> non-specific binding</li>
</ul></li>
<li class="fragment">Arrays included paired <em>mismatch</em> probes (MM) with a change at the 13th position
<ul>
<li class="fragment">Sometimes returned more signal than PM probes 🤪</li>
</ul></li>
<li class="fragment">Gene-level expression estimate obtained taking a robust average across probeset
<ul>
<li class="fragment">Model also included BG signal (but not always from MM probes)</li>
</ul></li>
</ul>
</section>
<section id="whole-transcript-arrays" class="slide level2">
<h2>Whole Transcript Arrays</h2>
<ul>
<li>Whole Transcript Arrays released by Affymetrix in mid-2000s
<ul>
<li>Marketed as Exon Arrays and Gene Arrays</li>
</ul></li>
<li>Probes along entire transcript <em>BUT</em> <span class="math inline">\(\leq\)</span> 4 probes/exon</li>
</ul>
<div class="fragment">
<div class="quarto-figure quarto-figure-left">
<figure>
<p><img data-src="transcriptomics/ArrayProbeLayout.png" style="width:70.0%"></p>
<figcaption>Source: Affymetrix Technical Note</figcaption>
</figure>
</div>
</div>
</section>
<section id="whole-transcript-arrays-1" class="slide level2 slide-only unlisted">
<h2>Whole Transcript Arrays</h2>
<aside class="notes">
<ul>
<li>Detecting alternate isoform usage on Exon Arrays was the focus of my PhD</li>
<li>My method was rubbish, but still better than anything else</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li>No successful methods for determining alternate isoform usage
<ul>
<li>Most people reverted back to gene-level signal</li>
<li>No real gains over 3’ Arrays beside more genes/array</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>RNA-Seq appeared at a similar time &amp; <strong>obliterated</strong> their market share
<ul>
<li>Alternate isoform usage in RNA-seq is <em>still</em> considered a bit exploratory</li>
</ul></li>
</ul>
</div>
</section></section>
<section>
<section id="microarray-analysis" class="title-slide slide level1 center">
<h1>Microarray Analysis</h1>

</section>
<section id="single-channel-data" class="slide level2">
<h2>Single Channel Data</h2>
<ul>
<li>We will have multiple arrays from each condition
<ul>
<li>Biological Replicates (hopefully <span class="math inline">\(\geq4\)</span> per condition)</li>
<li>Want to find changed expression in response to our biological hypothesis</li>
</ul></li>
</ul>
<ul>
<li class="fragment">Will the some arrays have higher/lower overall signal?
<ul>
<li class="fragment">Pipetting errors, hybridisation variability etc</li>
</ul></li>
<li class="fragment">Two Initial Problems to solve</li>
</ul>
<ol type="1">
<li class="fragment">Adjust for overall differences in signal <span class="fragment"><span class="math inline">\(\implies\)</span> Normalisation</span></li>
<li class="fragment">Removal of Background Signal (non-specific binding + optical noise) <br> <span class="fragment"><span class="math inline">\(\implies\)</span>Background Correction</span></li>
</ol>
</section>
<section id="normalisation" class="slide level2">
<h2>Normalisation</h2>
<div class="columns">
<div class="column" style="width:65%;">
<div class="quarto-figure quarto-figure-left">
<figure>
<p><img data-src="transcriptomics/quantileNorm.png"></p>
<figcaption>Example of raw PM probe intensities. Taken from <span class="citation" data-cites="pmid12538238">Bolstad et al. (<a href="#/references" role="doc-biblioref" onclick="">2003</a>)</span></figcaption>
</figure>
</div>
</div><div class="column" style="width:35%;">
<ul>
<li>The variation here is technical<br><span class="math inline">\(\implies\)</span> <em>not</em> due to biology</li>
<li>Higher variance reduces power of statistical testing</li>
<li>Can we reduce this?</li>
</ul>
<div class="fragment">
<ul>
<li>Quantile normalisation</li>
</ul>
</div>
</div></div>
</section>
<section id="normalisation-1" class="slide level2 slide-only unlisted">
<h2>Normalisation</h2>
<ul>
<li>Quantile normalisation is perfect for arrays with probes and probesets
<ul>
<li>Normalise probes, but estimate signal at the probeset level</li>
</ul></li>
</ul>
<ol type="1">
<li class="fragment">Find the lowest signal probe on each array</li>
<li class="fragment">Calculate the average signal across all arrays</li>
<li class="fragment">Give each of the probes the average signal</li>
<li class="fragment">Move to the next lowest signal probe until finished</li>
</ol>
<div class="fragment">
<ul>
<li>The lowest signal probes will be completely different across arrays</li>
<li>Effectively randomises noise</li>
<li>Leads to arrays with identical distributions</li>
</ul>
</div>
</section>
<section id="normalisation-2" class="slide level2 slide-only unlisted">
<h2>Normalisation</h2>
<div class="columns">
<div class="column">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="transcriptomics/preNorm.png"></p>
<figcaption>Before quantile normalisation</figcaption>
</figure>
</div>
</div><div class="column">
<div class="quarto-figure quarto-figure-center">
<figure>
<p><img data-src="transcriptomics/postNorm.png"></p>
<figcaption>After quantile normalisation<sup>1</sup></figcaption>
</figure>
</div>
</div></div>
<ul>
<li>Now we have identical distributions of signal across all arrays</li>
<li>Equivalent to having identical amounts of source material (mRNA)</li>
</ul>
<aside><ol class="aside-footnotes"><li id="fn3"><p>Images taken from: Bolstad, Probe Level Quantile Normalization for High Density Oligonucleotide Array Data Unpublished Manuscript, 2001</p></li></ol></aside></section>
<section id="background-correction" class="slide level2">
<h2>Background Correction</h2>
<ul>
<li>Background Correction performed <em>simultaneously with estimation of signal</em></li>
<li>Robust Multichip Average (RMA) <span class="citation" data-cites="rma2003">(<a href="#/references" role="doc-biblioref" onclick="">Irizarry et al. 2003</a>)</span>
<ul>
<li>Estimates signal for each array (<span class="math inline">\(\mu_i\)</span>)</li>
<li>Model includes probe affinities (<span class="math inline">\(\alpha_j\)</span>)</li>
<li>Doesn’t include MM probes</li>
<li>Fitted using robust statistics to reduce impact of outlier probes</li>
</ul></li>
</ul>
<p><span class="math display">\[
PM_{ij} = \mu_i + \alpha_j + \epsilon_{ij}
\]</span></p>
<div class="fragment">
<ul>
<li>Extended to GC-RMA <span class="citation" data-cites="wu2004model">(<a href="#/references" role="doc-biblioref" onclick="">Wu et al. 2004</a>)</span> to include GC content of probes</li>
</ul>
</div>
</section>
<section id="differential-expression-analysis" class="slide level2">
<h2>Differential Expression Analysis</h2>
<ul>
<li>A primary challenge is to detect where gene expression levels change in response to biological question
<ul>
<li>Often control samples Vs treated samples</li>
</ul></li>
<li>Microarray data is normally distributed on the log<sub>2</sub> scale
<ul>
<li>Can fit standard regression models</li>
</ul></li>
<li>For Treat Vs Control <span class="math inline">\(\implies\)</span> <span class="math inline">\(T\)</span>-test for each gene</li>
</ul>
<div class="fragment">
<p><span class="math display">\[
H_0: \text{No difference in average gene expression levels}\\
H_A: \text{Some difference in average gene expression levels}
\]</span></p>
<ul>
<li>NB: Experiments estimate the <em>true</em> expression level across a theoretical population</li>
</ul>
</div>
</section>
<section id="differential-expression-analysis-1" class="slide level2 slide-only unlisted">
<h2>Differential Expression Analysis</h2>
<ul>
<li>The Bioconductor package <code>limma</code> is the industry standard <span class="citation" data-cites="pmid16646809">(<a href="#/references" role="doc-biblioref" onclick="">Smyth 2004</a>)</span>
<ul>
<li>Still heavily used for modern RNA-Seq data</li>
<li>Models tailored to managing variances found in transcriptomic datasets</li>
</ul></li>
</ul>
<div class="fragment">
<ul>
<li>After testing <span class="math inline">\(\rightarrow\)</span> <span class="math inline">\(p\)</span>-value for each gene</li>
<li>Multiple testing becomes an issue (revise Steven Delean’s lecture)</li>
</ul>
</div>
</section>
<section id="multiple-testing" class="slide level2">
<h2>Multiple Testing</h2>
<ul>
<li>Researchers commonly use a <span class="math inline">\(p\)</span>-value <span class="math inline">\(&lt;0.05\)</span> to claim evidence of significant change
<ul>
<li>Often described as the rejection threshold <span class="math inline">\(\alpha = 0.05\)</span></li>
</ul></li>
</ul>
<blockquote>
<p>A <span class="math inline">\(p\)</span>-value represent the probability of observing data (i.e.&nbsp;evidence of change) as extreme, or more extreme than our data if <span class="math inline">\(H_0\)</span> is true</p>
</blockquote>
<div class="fragment">
<ul>
<li>i.e.&nbsp;how likely are we to see our evidence of change if nothing’s actually happening</li>
<li>Using <span class="math inline">\(\alpha = 0.05\)</span> we will randomly see evidence as strong as observed (or stronger) about 1 in 20 times, <em>if there’s really nothing happening</em></li>
</ul>
</div>
</section>
<section id="multiple-testing-1" class="slide level2 slide-only unlisted">
<h2>Multiple Testing</h2>
<ul>
<li>If testing across 10,000 genes <span class="math inline">\(\implies\)</span> 10,000 <span class="math inline">\(p\)</span>-values</li>
<li>If no change between groups in any gene (<span class="math inline">\(H_0\)</span> is always true) <br><span class="math inline">\(\implies\)</span> about 500 genes will return <span class="math inline">\(p&lt;0.05\)</span></li>
<li>These would be considered <em>false discoveries</em>, <em>false positives</em> or <em>Type I Errors</em>
<ul>
<li>False negatives are Type II errors</li>
</ul></li>
</ul>
<table class="caption-top">
<thead>
<tr class="header">
<th style="text-align: left;"></th>
<th><span class="math inline">\(H_0\)</span> True</th>
<th><span class="math inline">\(H_0\)</span> False</th>
</tr>
</thead>
<tbody>
<tr class="odd">
<td style="text-align: left;"><strong>Accept <span class="math inline">\(H_0\)</span></strong></td>
<td>✅</td>
<td>Type II Error</td>
</tr>
<tr class="even">
<td style="text-align: left;"><strong>Reject <span class="math inline">\(H_0\)</span></strong></td>
<td><span class="text-red">Type I Error</span></td>
<td>✅</td>
</tr>
</tbody>
</table>
</section>
<section id="multiple-testing-2" class="slide level2 slide-only unlisted">
<h2>Multiple Testing</h2>
<ul>
<li>Reducing false positives is vital <span class="math inline">\(\implies\)</span> don’t want to waste research $$$</li>
<li>An intuitive solution would be to use a lower value for <span class="math inline">\(\alpha\)</span>
<ul>
<li>Using <span class="math inline">\(\alpha = 0.01\)</span> <span class="math inline">\(\implies\)</span> 1 in 100 random <span class="math inline">\(p\)</span>-values under <span class="math inline">\(H_0\)</span><br><span class="math inline">\(\implies\)</span> only 100 false discoveries in 10,000 genes</li>
<li>But this makes it harder for <em>true positives</em> to be identified <br><span class="math inline">\(\implies\)</span> more Type II errors (false negatives)</li>
</ul></li>
<li>No perfect solution for minimising Type I errors without increasing Type II errors</li>
</ul>
</section>
<section id="the-bonferroni-correction" class="slide level2">
<h2>The Bonferroni Correction</h2>
<ul>
<li>A commonly used approach is the Bonferroni Correction</li>
<li><span class="math inline">\(\alpha\)</span> is divided by the number of tests <br><span class="math inline">\(\implies\)</span> probability of one false discovery remains at the original <span class="math inline">\(\alpha\)</span>
<ul>
<li>Known as Family Wise Error Rate (FWER) control</li>
</ul></li>
</ul>
<ul>
<li class="fragment">For <span class="math inline">\(n=10,000\)</span> genes we would use <span class="math inline">\(\alpha^* = \frac{0.05}{n} = 5\times10^{-6}\)</span></li>
<li class="fragment">The equivalent is to multiply all <span class="math inline">\(p\)</span>-values by <span class="math inline">\(n\)</span> to give adjusted <span class="math inline">\(p\)</span>-values</li>
<li class="fragment"><span class="math inline">\(p_{adj} = \min(1, p \times n)\)</span> ensures <span class="math inline">\(p\)</span>-values are always <span class="math inline">\(\leq1\)</span>
<ul>
<li class="fragment">Referred to as Bonferroni-adjusted <span class="math inline">\(p\)</span>-values</li>
</ul></li>
</ul>
</section>
<section id="the-false-discovery-rate" class="slide level2">
<h2>The False Discovery Rate</h2>
<ul>
<li>An alternative is to allow a set proportion of your results to be <em>false positives</em></li>
<li>Instead of controlling the FWER, we control the False Discovery Rate (FDR)
<ul>
<li>Using <span class="math inline">\(\alpha = 0.05\)</span> <span class="math inline">\(\implies\)</span> 95% of results will be appropriate rejections of <span class="math inline">\(H_0\)</span></li>
<li>Is an estimate of the true FDR</li>
<li>Effectively allows a small amount of noise into our data</li>
<li>For larger network-style or downstream analysis <span class="math inline">\(\implies\)</span> signal drowns out noise</li>
</ul></li>
<li>Is almost always more powerful that the Bonferroni correction
<ul>
<li>More power <span class="math inline">\(\implies\)</span> more ability to find the <em>true positives</em></li>
</ul></li>
</ul>
</section>
<section id="the-false-discovery-rate-1" class="slide level2 slide-only unlisted">
<h2>The False Discovery Rate</h2>
<aside class="notes">
<ul>
<li>The adjustment algorithm is a bit beyond this course</li>
</ul>
<style type="text/css">
        span.MJX_Assistive_MathML {
          position:absolute!important;
          clip: rect(1px, 1px, 1px, 1px);
          padding: 1px 0 0 0!important;
          border: 0!important;
          height: 1px!important;
          width: 1px!important;
          overflow: hidden!important;
          display:block!important;
      }</style></aside>
<ul>
<li>Most common FDR method is the Benjamini-Hochberg adjustment <span class="citation" data-cites="BenjaminiHochbergFDR">(<a href="#/references" role="doc-biblioref" onclick="">Benjamini and Hochberg 1995</a>)</span></li>
</ul>
<ol type="1">
<li class="fragment"><span class="math inline">\(p\)</span>-values are given rank <span class="math inline">\(i = 1, 2, \ldots, n\)</span> where we have <span class="math inline">\(n\)</span> tests (i.e.&nbsp;genes)</li>
<li class="fragment">Find maximum value of <span class="math inline">\(i\)</span> where <span class="math inline">\(p_i \leq \alpha \frac{i}{n}\)</span></li>
</ol>
<ul>
<li class="fragment">FDR-adjusted <span class="math inline">\(p\)</span>-values easily obtained using <code>R</code>
<ul>
<li class="fragment"><code>p.adjust(p, method = "BH")</code></li>
</ul></li>
</ul>
</section>
<section id="closing-comments" class="slide level2">
<h2>Closing Comments</h2>
<ul>
<li>Microarray signal estimates follow a normal distribution (<span class="math inline">\(\mathcal{N}(\mu, \sigma)\)</span>) with log-transformed</li>
<li>We can apply linear regression models
<ul>
<li>In a simple A vs B experiment <span class="math inline">\(\equiv\)</span> <span class="math inline">\(T\)</span>-tests</li>
</ul></li>
<li>Estimates of change from DE analysis often referred to as logFC (log fold-change)</li>
<li><span class="math inline">\(p\)</span>-values are usually FDR-adjusted
<ul>
<li>Gives best compromise of power vs error-rate control</li>
</ul></li>
</ul>
</section>
<section id="closing-comments-1" class="slide level2 slide-only unlisted">
<h2>Closing Comments</h2>
<ul>
<li>Foundations built during the microarray era enabled analysis of RNA-Seq data
<ul>
<li>RNA-Seq data is not normally distributed</li>
</ul></li>
<li>Core principles and methods developed during this era still apply
<ul>
<li>Normalisation, DE analysis, multiple testing etc</li>
</ul></li>
<li>Many bioinformaticians from the microarray era are still very active</li>
<li>A lot of development occurred in Australia (e.g.&nbsp;Prof Gordon Smyth, WEHI)
<ul>
<li>Next generation have been trained &amp; mentored at WEHI, USyd etc</li>
</ul></li>
</ul>
</section>
<section id="references" class="slide level2 smaller scrollable">
<h2>References</h2>

<div id="refs" class="references csl-bib-body hanging-indent" data-entry-spacing="0" role="list">
<div id="ref-Abugessaisa2021-fn" class="csl-entry" role="listitem">
Abugessaisa, Imad, Jordan A Ramilowski, Marina Lizio, Jesicca Severin, Akira Hasegawa, Jayson Harshbarger, Atsushi Kondo, et al. 2020. <span>“FANTOM Enters 20th Year: Expansion of Transcriptomic Atlases and Functional Annotation of Non-Coding RNAs.”</span> <em>Nucleic Acids Research</em> 49 (D1): D892–98. <a href="https://doi.org/10.1093/nar/gkaa1054">https://doi.org/10.1093/nar/gkaa1054</a>.
</div>
<div id="ref-1991VenterEST" class="csl-entry" role="listitem">
Adams, Mark D., Jenny M. Kelley, Jeannine D. Gocayne, Mark Dubnick, Mihael H. Polymeropoulos, Hong Xiao, Carl R. Merril, et al. 1991. <span>“Complementary DNA Sequencing: Expressed Sequence Tags and Human Genome Project.”</span> <em>Science</em> 252 (5013): 1651–56. <a href="http://www.jstor.org/stable/2876333">http://www.jstor.org/stable/2876333</a>.
</div>
<div id="ref-1977NorthernBlot" class="csl-entry" role="listitem">
Alwine, J. C., D. J. Kemp, and G. R. Stark. 1977. <span>“<span class="nocase"><span>M</span>ethod for detection of specific <span>R</span><span>N</span><span>A</span>s in agarose gels by transfer to diazobenzyloxymethyl-paper and hybridization with <span>D</span><span>N</span><span>A</span> probes</span>.”</span> <em>Proc. Natl. Acad. Sci. U.S.A.</em> 74 (12): 5350–54.
</div>
<div id="ref-BenjaminiHochbergFDR" class="csl-entry" role="listitem">
Benjamini, Yoav, and Yosef Hochberg. 1995. <span>“Controlling the False Discovery Rate: A Practical and Powerful Approach to Multiple Testing.”</span> <em>Journal of the Royal Statistical Society. Series B (Methodological)</em> 57 (1): 289–300. <a href="http://www.jstor.org/stable/2346101">http://www.jstor.org/stable/2346101</a>.
</div>
<div id="ref-pmid12538238" class="csl-entry" role="listitem">
Bolstad, B. M., R. A. Irizarry, M. Astrand, and T. P. Speed. 2003. <span>“<span class="nocase"><span>A</span> comparison of normalization methods for high density oligonucleotide array data based on variance and bias</span>.”</span> <em>Bioinformatics</em> 19 (2): 185–93.
</div>
<div id="ref-dudoit02-ma" class="csl-entry" role="listitem">
Dudoit, Sandrine, Yee Hwa Yang, Matthew J. Callow, and Terence P. Speed. 2002. <span>“STATISTICAL METHODS FOR IDENTIFYING DIFFERENTIALLY EXPRESSED GENES IN REPLICATED cDNA MICROARRAY EXPERIMENTS.”</span> <em>Statistica Sinica</em> 12 (1): 111–39. <a href="http://www.jstor.org/stable/24307038">http://www.jstor.org/stable/24307038</a>.
</div>
<div id="ref-Fang2015-uq" class="csl-entry" role="listitem">
Fang, Rui, Walter N Moss, Michael Rutenberg-Schoenberg, and Matthew D Simon. 2015. <span>“Probing Xist <span>RNA</span> Structure in Cells Using Targeted Structure-Seq.”</span> <em>PLoS Genet.</em> 11 (12): e1005668.
</div>
<div id="ref-rma2003" class="csl-entry" role="listitem">
Irizarry, Rafael A., Bridget Hobbs, Francois Collin, Yasmin D. Beazer‐Barclay, Kristen J. Antonellis, Uwe Scherf, and Terence P. Speed. 2003. <span>“Exploration, Normalization, and Summaries of High Density Oligonucleotide Array Probe Level Data.”</span> <em>Biostatistics</em> 4 (2): 249–64. <a href="https://doi.org/10.1093/biostatistics/4.2.249">https://doi.org/10.1093/biostatistics/4.2.249</a>.
</div>
<div id="ref-Shafee2017-zv" class="csl-entry" role="listitem">
Shafee, Thomas, and Rohan Lowe. 2017. <span>“Eukaryotic and Prokaryotic Gene Structure.”</span> <em><span>WikiJournal</span> of Medicine</em>, January. <a href="https://doi.org/10.15347/WJM/2017.002">https://doi.org/10.15347/WJM/2017.002</a>.
</div>
<div id="ref-Shalon01071996" class="csl-entry" role="listitem">
Shalon, D, S J Smith, and P O Brown. 1996. <span>“A DNA Microarray System for Analyzing Complex DNA Samples Using Two-Color Fluorescent Probe Hybridization.”</span> <em>Genome Research</em> 6 (7): 639–45. <a href="https://doi.org/10.1101/gr.6.7.639">https://doi.org/10.1101/gr.6.7.639</a>.
</div>
<div id="ref-pmid16646809" class="csl-entry" role="listitem">
Smyth, G. K. 2004. <span>“<span class="nocase"><span>L</span>inear models and empirical bayes methods for assessing differential expression in microarray experiments</span>.”</span> <em>Stat Appl Genet Mol Biol</em> 3: Article3.
</div>
<div id="ref-pmid7570003" class="csl-entry" role="listitem">
Velculescu, V. E., L. Zhang, B. Vogelstein, and K. W. Kinzler. 1995. <span>“<span class="nocase"><span>S</span>erial analysis of gene expression</span>.”</span> <em>Science</em> 270 (5235): 484–87.
</div>
<div id="ref-Wang2009-hf" class="csl-entry" role="listitem">
Wang, Zhong, Mark Gerstein, and Michael Snyder. 2009. <span>“<span>RNA</span>-Seq: A Revolutionary Tool for Transcriptomics.”</span> <em>Nat. Rev. Genet.</em> 10 (1): 57–63.
</div>
<div id="ref-wu2004model" class="csl-entry" role="listitem">
Wu, Zhijin, Rafael A Irizarry, Robert Gentleman, Francisco Martinez-Murillo, and Forrest Spencer. 2004. <span>“A Model-Based Background Adjustment for Oligonucleotide Expression Arrays.”</span> <em>Journal of the American Statistical Association</em> 99 (468): 909–17.
</div>
</div>
</section></section>

    </div>
  <div class="quarto-auto-generated-content" style="display: none;">
<div class="footer footer-default">

</div>
</div></div>

  <script>window.backupDefine = window.define; window.define = undefined;</script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/dist/reveal.js"></script>
  <!-- reveal.js plugins -->
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/quarto-line-highlight/line-highlight.js"></script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/pdf-export/pdfexport.js"></script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/reveal-menu/menu.js"></script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/reveal-menu/quarto-menu.js"></script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/quarto-support/support.js"></script>
  

  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/notes/notes.js"></script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/search/search.js"></script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/zoom/zoom.js"></script>
  <script src="transcriptomics-lecture1-slides_files/libs/revealjs/plugin/math/math.js"></script>
  <script>window.define = window.backupDefine; window.backupDefine = undefined;</script>

  <script>

      // Full list of configuration options available at:
      // https://revealjs.com/config/
      Reveal.initialize({
'controlsAuto': true,
'previewLinksAuto': false,
'pdfSeparateFragments': false,
'autoAnimateEasing': "ease",
'autoAnimateDuration': 1,
'autoAnimateUnmatched': true,
'jumpToSlide': true,
'menu': {"side":"left","useTextContentForMissingTitles":true,"markers":false,"loadIcons":false,"custom":[{"title":"Tools","icon":"<i class=\"fas fa-gear\"></i>","content":"<ul class=\"slide-menu-items\">\n<li class=\"slide-tool-item active\" data-item=\"0\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.fullscreen(event)\"><kbd>f</kbd> Fullscreen</a></li>\n<li class=\"slide-tool-item\" data-item=\"1\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.speakerMode(event)\"><kbd>s</kbd> Speaker View</a></li>\n<li class=\"slide-tool-item\" data-item=\"2\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.overview(event)\"><kbd>o</kbd> Slide Overview</a></li>\n<li class=\"slide-tool-item\" data-item=\"3\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.togglePdfExport(event)\"><kbd>e</kbd> PDF Export Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"4\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.toggleScrollView(event)\"><kbd>r</kbd> Scroll View Mode</a></li>\n<li class=\"slide-tool-item\" data-item=\"5\"><a href=\"#\" onclick=\"RevealMenuToolHandlers.keyboardHelp(event)\"><kbd>?</kbd> Keyboard Help</a></li>\n</ul>"}],"openButton":true},
'smaller': false,
 
        // Display controls in the bottom right corner
        controls: false,

        // Help the user learn the controls by providing hints, for example by
        // bouncing the down arrow when they first encounter a vertical slide
        controlsTutorial: false,

        // Determines where controls appear, "edges" or "bottom-right"
        controlsLayout: 'edges',

        // Visibility rule for backwards navigation arrows; "faded", "hidden"
        // or "visible"
        controlsBackArrows: 'faded',

        // Display a presentation progress bar
        progress: true,

        // Display the page number of the current slide
        slideNumber: 'c',

        // 'all', 'print', or 'speaker'
        showSlideNumber: 'all',

        // Add the current slide number to the URL hash so that reloading the
        // page/copying the URL will return you to the same slide
        hash: true,

        // Start with 1 for the hash rather than 0
        hashOneBasedIndex: false,

        // Flags if we should monitor the hash and change slides accordingly
        respondToHashChanges: true,

        // Push each slide change to the browser history
        history: true,

        // Enable keyboard shortcuts for navigation
        keyboard: true,

        // Enable the slide overview mode
        overview: true,

        // Disables the default reveal.js slide layout (scaling and centering)
        // so that you can use custom CSS layout
        disableLayout: false,

        // Vertical centering of slides
        center: false,

        // Enables touch navigation on devices with touch input
        touch: true,

        // Loop the presentation
        loop: false,

        // Change the presentation direction to be RTL
        rtl: false,

        // see https://revealjs.com/vertical-slides/#navigation-mode
        navigationMode: 'linear',

        // Randomizes the order of slides each time the presentation loads
        shuffle: false,

        // Turns fragments on and off globally
        fragments: true,

        // Flags whether to include the current fragment in the URL,
        // so that reloading brings you to the same fragment position
        fragmentInURL: false,

        // Flags if the presentation is running in an embedded mode,
        // i.e. contained within a limited portion of the screen
        embedded: false,

        // Flags if we should show a help overlay when the questionmark
        // key is pressed
        help: true,

        // Flags if it should be possible to pause the presentation (blackout)
        pause: true,

        // Flags if speaker notes should be visible to all viewers
        showNotes: false,

        // Global override for autoplaying embedded media (null/true/false)
        autoPlayMedia: null,

        // Global override for preloading lazy-loaded iframes (null/true/false)
        preloadIframes: null,

        // Number of milliseconds between automatically proceeding to the
        // next slide, disabled when set to 0, this value can be overwritten
        // by using a data-autoslide attribute on your slides
        autoSlide: 0,

        // Stop auto-sliding after user input
        autoSlideStoppable: true,

        // Use this method for navigation when auto-sliding
        autoSlideMethod: null,

        // Specify the average time in seconds that you think you will spend
        // presenting each slide. This is used to show a pacing timer in the
        // speaker view
        defaultTiming: null,

        // Enable slide navigation via mouse wheel
        mouseWheel: false,

        // The display mode that will be used to show slides
        display: 'block',

        // Hide cursor if inactive
        hideInactiveCursor: true,

        // Time before the cursor is hidden (in ms)
        hideCursorTime: 5000,

        // Opens links in an iframe preview overlay
        previewLinks: false,

        // Transition style (none/fade/slide/convex/concave/zoom)
        transition: 'none',

        // Transition speed (default/fast/slow)
        transitionSpeed: 'default',

        // Transition style for full page slide backgrounds
        // (none/fade/slide/convex/concave/zoom)
        backgroundTransition: 'none',

        // Number of slides away from the current that are visible
        viewDistance: 3,

        // Number of slides away from the current that are visible on mobile
        // devices. It is advisable to set this to a lower number than
        // viewDistance in order to save resources.
        mobileViewDistance: 2,

        // The "normal" size of the presentation, aspect ratio will be preserved
        // when the presentation is scaled to fit different resolutions. Can be
        // specified using percentage units.
        width: 1280,

        height: 720,

        // Factor of the display size that should remain empty around the content
        margin: 0.1,

        math: {
          mathjax: 'https://cdnjs.cloudflare.com/ajax/libs/mathjax/2.7.9/MathJax.js',
          config: 'TeX-AMS_HTML-full',
          tex2jax: {
            inlineMath: [['\\(','\\)']],
            displayMath: [['\\[','\\]']],
            balanceBraces: true,
            processEscapes: false,
            processRefs: true,
            processEnvironments: true,
            preview: 'TeX',
            skipTags: ['script','noscript','style','textarea','pre','code'],
            ignoreClass: 'tex2jax_ignore',
            processClass: 'tex2jax_process'
          },
        },

        // reveal.js plugins
        plugins: [QuartoLineHighlight, PdfExport, RevealMenu, QuartoSupport,

          RevealMath,
          RevealNotes,
          RevealSearch,
          RevealZoom
        ]
      });
    </script>
    <div id="logo-tki" style="position: fixed; z-index: 1000;">
      <img src="https://www.thekids.org.au/Static/Theme/images/TheKids-RGB_Logo_Colour-Pattern.svg" style="height: 80px;">
    </div>
    <div id="logo-bodl" style="position: fixed; z-index: 1000;">
      <img src="https://raw.githubusercontent.com/black-ochre-data-labs/20250902_ASI_RWorkshop/refs/heads/main/assets/bodl_logo_white_background.jpg" style="height: 80px;">
    </div>
    <div id="logo-uoa" style="z-index: 1000;">
      <img src="transcriptomics/UoA_logo_col_vert.png" style="height: 80px;">
    </div>


    <script>
    function positionLogos() {
      const tkiLogo = document.getElementById('logo-tki');
      const uoaLogo = document.getElementById('logo-uoa');
      const bodlLogo = document.getElementById('logo-bodl');

      // Check if this is RevealJS
      const isRevealJS = document.body.classList.contains('reveal') ||
                         document.querySelector('.reveal') !== null ||
                         typeof Reveal !== 'undefined';

      if (isRevealJS) {
        // RevealJS - both logos at bottom
        tkiLogo.style.bottom = '10px';
        tkiLogo.style.right = '340px';
        tkiLogo.style.top = 'auto';
        tkiLogo.style.display = 'block';

        bodlLogo.style.bottom = '10px';
        bodlLogo.style.right = '160px';
        bodlLogo.style.top = 'auto';
        bodlLogo.style.display = 'block';

        uoaLogo.style.bottom = '10px';
        uoaLogo.style.right = '40px';
        uoaLogo.style.top = 'auto';
        uoaLogo.style.display = 'block';
        uoaLogo.style.position = 'fixed';
      } else {
        // Flat HTML - only right logo at top
        tkiLogo.style.display = 'none';
        bodlLogo.style.display = 'none';

        uoaLogo.style.top = '30px';
        uoaLogo.style.left = '60%';
        uoaLogo.style.bottom = 'auto';
        uoaLogo.style.position = 'absolute';
        uoaLogo.style.display = 'block';
      }
    }

    // Run positioning
    document.addEventListener('DOMContentLoaded', positionLogos);
    setTimeout(positionLogos, 100);
    </script>
    <script id="quarto-html-after-body" type="application/javascript">
      window.document.addEventListener("DOMContentLoaded", function (event) {
        const tabsets =  window.document.querySelectorAll(".panel-tabset-tabby")
        tabsets.forEach(function(tabset) {
          const tabby = new Tabby('#' + tabset.id);
        });
        const isCodeAnnotation = (el) => {
          for (const clz of el.classList) {
            if (clz.startsWith('code-annotation-')) {                     
              return true;
            }
          }
          return false;
        }
        const onCopySuccess = function(e) {
          // button target
          const button = e.trigger;
          // don't keep focus
          button.blur();
          // flash "checked"
          button.classList.add('code-copy-button-checked');
          var currentTitle = button.getAttribute("title");
          button.setAttribute("title", "Copied!");
          let tooltip;
          if (window.bootstrap) {
            button.setAttribute("data-bs-toggle", "tooltip");
            button.setAttribute("data-bs-placement", "left");
            button.setAttribute("data-bs-title", "Copied!");
            tooltip = new bootstrap.Tooltip(button, 
              { trigger: "manual", 
                customClass: "code-copy-button-tooltip",
                offset: [0, -8]});
            tooltip.show();    
          }
          setTimeout(function() {
            if (tooltip) {
              tooltip.hide();
              button.removeAttribute("data-bs-title");
              button.removeAttribute("data-bs-toggle");
              button.removeAttribute("data-bs-placement");
            }
            button.setAttribute("title", currentTitle);
            button.classList.remove('code-copy-button-checked');
          }, 1000);
          // clear code selection
          e.clearSelection();
        }
        const getTextToCopy = function(trigger) {
            const codeEl = trigger.previousElementSibling.cloneNode(true);
            for (const childEl of codeEl.children) {
              if (isCodeAnnotation(childEl)) {
                childEl.remove();
              }
            }
            return codeEl.innerText;
        }
        const clipboard = new window.ClipboardJS('.code-copy-button:not([data-in-quarto-modal])', {
          text: getTextToCopy
        });
        clipboard.on('success', onCopySuccess);
        if (window.document.getElementById('quarto-embedded-source-code-modal')) {
          const clipboardModal = new window.ClipboardJS('.code-copy-button[data-in-quarto-modal]', {
            text: getTextToCopy,
            container: window.document.getElementById('quarto-embedded-source-code-modal')
          });
          clipboardModal.on('success', onCopySuccess);
        }
          var localhostRegex = new RegExp(/^(?:http|https):\/\/localhost\:?[0-9]*\//);
          var mailtoRegex = new RegExp(/^mailto:/);
            var filterRegex = new RegExp('/' + window.location.host + '/');
          var isInternal = (href) => {
              return filterRegex.test(href) || localhostRegex.test(href) || mailtoRegex.test(href);
          }
          // Inspect non-navigation links and adorn them if external
         var links = window.document.querySelectorAll('a[href]:not(.nav-link):not(.navbar-brand):not(.toc-action):not(.sidebar-link):not(.sidebar-item-toggle):not(.pagination-link):not(.no-external):not([aria-hidden]):not(.dropdown-item):not(.quarto-navigation-tool):not(.about-link)');
          for (var i=0; i<links.length; i++) {
            const link = links[i];
            if (!isInternal(link.href)) {
              // undo the damage that might have been done by quarto-nav.js in the case of
              // links that we want to consider external
              if (link.dataset.originalHref !== undefined) {
                link.href = link.dataset.originalHref;
              }
            }
          }
        function tippyHover(el, contentFn, onTriggerFn, onUntriggerFn) {
          const config = {
            allowHTML: true,
            maxWidth: 500,
            delay: 100,
            arrow: false,
            appendTo: function(el) {
                return el.closest('section.slide') || el.parentElement;
            },
            interactive: true,
            interactiveBorder: 10,
            theme: 'light-border',
            placement: 'bottom-start',
          };
          if (contentFn) {
            config.content = contentFn;
          }
          if (onTriggerFn) {
            config.onTrigger = onTriggerFn;
          }
          if (onUntriggerFn) {
            config.onUntrigger = onUntriggerFn;
          }
            config['offset'] = [0,0];
            config['maxWidth'] = 700;
          window.tippy(el, config); 
        }
        const noterefs = window.document.querySelectorAll('a[role="doc-noteref"]');
        for (var i=0; i<noterefs.length; i++) {
          const ref = noterefs[i];
          tippyHover(ref, function() {
            // use id or data attribute instead here
            let href = ref.getAttribute('data-footnote-href') || ref.getAttribute('href');
            try { href = new URL(href).hash; } catch {}
            const id = href.replace(/^#\/?/, "");
            const note = window.document.getElementById(id);
            if (note) {
              return note.innerHTML;
            } else {
              return "";
            }
          });
        }
        const findCites = (el) => {
          const parentEl = el.parentElement;
          if (parentEl) {
            const cites = parentEl.dataset.cites;
            if (cites) {
              return {
                el,
                cites: cites.split(' ')
              };
            } else {
              return findCites(el.parentElement)
            }
          } else {
            return undefined;
          }
        };
        var bibliorefs = window.document.querySelectorAll('a[role="doc-biblioref"]');
        for (var i=0; i<bibliorefs.length; i++) {
          const ref = bibliorefs[i];
          const citeInfo = findCites(ref);
          if (citeInfo) {
            tippyHover(citeInfo.el, function() {
              var popup = window.document.createElement('div');
              citeInfo.cites.forEach(function(cite) {
                var citeDiv = window.document.createElement('div');
                citeDiv.classList.add('hanging-indent');
                citeDiv.classList.add('csl-entry');
                var biblioDiv = window.document.getElementById('ref-' + cite);
                if (biblioDiv) {
                  citeDiv.innerHTML = biblioDiv.innerHTML;
                }
                popup.appendChild(citeDiv);
              });
              return popup.innerHTML;
            });
          }
        }
      });
      </script>
    

</body></html>