---
title: "Transcriptomics: Lecture 1"
subtitle: |
  | Biotech 7005/Bioinf 3000
  | Frontiers of Biotechnology: Bioinformatics and Systems Modelling
  | The University of Adelaide
author: "Dr Stevie Pederson (They/Them)<br>stevie.pederson@thekids.org.au"
email: "stevie.pederson@thekids.org.au"
institute: |
  | Black Ochre Data Labs
  | The Kids Research Institute Australia
date: "2025-09-01"
date-format: "long"
bibliography: transcriptomics/references.bib
format: 
  revealjs:
    theme: [transcriptomics/custom.scss]
    height: 720
    width: 1280
    sansfont: Times New Roman
    slide-number: c
    show-slide-number: all
    include-after-body: transcriptomics/uoa_bodl_logos.html
  html: 
    css: [transcriptomics/custom.scss]
    output-file: transcriptomics1-lecture1.html
    embed-resources: true    
    toc: true
    toc-depth: 2
    include-in-header: transcriptomics/uoa_bodl_logos.html
---

## Helpful Links

- The slides are available [here](./transcriptomics-lecture1-slides.html)
- The same notes are available as a single-page html [here](./transcriptomics1-lecture1.html)


## [Welcome To Country]{.text-red}

::: {.text-red}

*I'd like to acknowledge the Kaurna people as the traditional owners and custodians of the land we know today as the Adelaide Plains, where I live & work.*
<br><br>
*I also acknowledge the deep feelings of attachment and relationship of the Kaurna people to their place.*
<br><br>
*I pay my respects to the cultural authority of Aboriginal and Torres Strait Islander peoples from other areas of Australia, and pay my respects to Elders past, present and emerging, and acknowledge any Aboriginal Australians who may be with us today*

:::

# Introduction To Transcriptomics

## Introduction

- Postdoctoral Bioinformatician, Black Ochre Data Labs, Adelaide
- Working in collaboration with members of the SA Aboriginal community
- Multi-omics project to identify and address the underlying causes of high T2D rates and complications
    + Using genomics, epigenomics, transcriptomics and other layers
    + My focus is on the transcriptomics layer


## Why Transcriptomics?

:::: {.columns}

::: {.column}

- DNA can be described as being like a giant book of instructions

:::{.incremental}
- Some regions are defined as *genes*
    + Originally considered to be the basic unit of inheritance
    + Now commonly used to describe a region of DNA transcribed into RNA
    + Recent discovery of enhancer-RNA (eRNA) muddies the water a little

:::
:::
::: {.column}


::: {style="font-size: 60%; text-align: center;"}

![](transcriptomics/DNA_to_protein_or_ncRNA.png)
<br>By <a href="//commons.wikimedia.org/wiki/User:Evolution_and_evolvability" title="User:Evolution and evolvability">Thomas Shafee</a> - <span class="int-own-work" lang="en">Own work</span>, <a href="https://creativecommons.org/licenses/by/4.0" title="Creative Commons Attribution 4.0">CC BY 4.0</a>, <a href="https://commons.wikimedia.org/w/index.php?curid=39441809">Wikimedia Link</a>

:::
:::

::::

## Why Transcriptomics? {.slide-only .unlisted}

- DNA $\rightarrow$ mRNA $\rightarrow$ Proteins
    + Commonly referred to as the Central Dogma of Biology
- Proteins are the *workhorses* of the cell & body
    + Do most of the work, and are responsible for most of the structure
    + Examples like keratin (hair), haemoglobin (oxygen transport) etc
  
::: {.fragment}
  
- ncRNAs are also highly functional
    + Ribosomal RNA (rRNA) + tRNA $\rightarrow$ translation from mRNA to Protein
    + microRNAs play a role in gene-regulation via mRNA stability
    
:::

## Why Transcriptomics? {.slide-only .unlisted}

:::: {.columns}

::: {.column}
![Taken from @Fang2015-uq](transcriptomics/pgen.1005668.g006.png)
:::

::: {.column}
- Most RNA is single-stranded but can have extremely complex structure
    + Shown is a 2kb region from the lncRNA Xist (17kb in total)
- Coats the *entire X chromosome* during X inactivation
- Also interacts with the antisense lncRNA Tsix

:::

::::

## Why Transcriptomics? {.slide-only .unlisted}

### Definition

Based on @Wang2009-hf

> The transcriptome can be defined as the complete set of (RNA) transcripts in a cell, or a population of cells, for a specific developmental stage or physiological condition 


::: {.fragment}
- Transcriptomics is simply the study of the transcriptome
- Can be the *entire RNA content of a cell* (or cells) or a subset of molecules (e.g. mRNA, miRNA)
:::



## Why Transcriptomics? {.slide-only .unlisted}

- Is a snapshot of the **dynamic biological processes** associated with a biological question
- Use to make inference about these processes
    + Identify therapeutic targets for Cardiovascular Disease
    + Biomarkers for CAR-T cells
    + Key drivers of correlated gene networks
    + Early drivers of neurodegeneration in Alzheimer's
- Assumed to be low-level
    + DNA $\rightarrow$ [RNA]{.text-red} [$\rightarrow$ Protein $\rightarrow$ Metabolites, Signalling molecules, etc ...]{.fragment}

<!-- ## Why Transcriptomics? {.slide-only .unlisted} -->

<!-- - Is the lowest molecular level where quantity becomes a key aspect -->
<!--     + Highly-expressed, or low-expressed genes are important -->
<!--     + Changes in response to stimulus impact gene expression levels -->

<!-- ::: {.fragment} -->
<!-- - Much of the early transcriptomic analyses were quantitative -->
<!--     + Sequence variation often captured at DNA-level -->
<!-- - Now extending to transcript structure and modifications -->
<!--     + Identification of fusion transcripts, RNA-methylation etc -->

<!-- ::: -->

## Why Transcriptomics? {.slide-only .unlisted}

- Early techniques were often using large numbers of cells
    + Often multiple cell types within a biological sample
- Modern techniques are incredibly detailed
    + Single-Cell RNA characterises exact cell types and cell trajectories
    + Spatial transcriptomics used to identify co-located cells in tissue
    + Identify cell-cell signalling *in situ*

## What Is Transcription

:::: {.columns}

::: {.column width='40%'}

### Definition

> Transcription is the process of making an RNA copy of a gene sequence

:::

::: {.column width='60%'}

::: {style="font-size: 90%; text-align: center;"}

![Figure taken from [^1] Licensed under CC-BY 4.0 by OpenStax](transcriptomics/transcription.jpg)

:::

:::

::::

[^1]: https://openoregon.pressbooks.pub/mhccbiology102/chapter/transcription/

## Steps of Transcription 


1. **RNA polymerase binds to the promoter** along with $\geq1$ transcription factors

::: {.incremental}
2. **RNA polymerase creates a transcription bubble**
    + separates the two DNA strands, breaking hydrogen bonds between complementary DNA nucleotides.
3. **RNA polymerase adds RNA nucleotides**
    + complementary to the antisense DNA strand.
4. **RNA sugar-phosphate backbone forms**
5. **Hydrogen bonds of the RNA–DNA complex break** freeing the newly synthesized RNA strand.

:::

## Steps of Transcription {.slide-only .unlisted}

#### If the cell is a eukaryotic cell

6. **RNA processing**
    + This may include polyadenylation, capping and splicing
    + Occurs during (or immediately after) transcription
7. **RNA Localisation**
    + The RNA may remain in the nucleus or exit to the cytoplasm through the nuclear pore complex

::: {.fragment}
- Eukaryotic mRNA, miRNA & snRNA transcription uses *RNA Polymerase II*
    + RNA Pol I: rRNA
    + RNA Pol III: tRNA, 5S RNA some small RNAs
    
:::

## Eukaryotic mRNA Processing

- Nuclear mRNA have 5' cap added
    + Protects single-stranded mRNA from degradation
    + Regulates nuclear export
    + Promotes translation into protein
    
::: {.fragment}
- mRNAs are polyadenylated at the 3' end
    + Also protects from degradation
    + Aids in transcription termination, export and translation
    
:::
::: {.fragment}
- Introns are spliced out as required    
:::


## Eukaryotic mRNA Processing {.slide-only .unlisted}

![Taken from @Shafee2017-zv](transcriptomics/eukaryotic_mrna_processing.svg)

## Alternate Transcripts and Isoforms


![Image by the National Human Genome Research Institute](transcriptomics/DNA_alternative_splicing.png)

## Transcriptome Resources

- Reference Transcriptomes & Genomes are now commonly available
    + Incorporate experimentally derived & predicted sequences + loci
- Gencode[^2] provide highest quality for mouse & human
    + Release 48 (GRCh38): 78,686 genes + 385,669 transcripts

::: {.fragment}
- Other organisms from Ensembl, RefSeq, UCSC etc
     + Zebrafish, Rat, Chicken, Drosophila, Wheat, Yeast, *E. Coli* etc

:::

::: {.fragment}
- Sometimes we build novel transcriptomes from specific tissues
    + e.g. sea snake venom gland, shiraz fruit
    
:::


[^2]: https://www.gencodegenes.org/

# Early Transcriptomics

## Northern Blotting

- Northern blot [@1977NorthernBlot] extended DNA-based methods (i.e Southern blot) [$\implies$ Earliest single-gene method]{.fragment}

::: {.fragment}

- Gel Electrophoresis then hybridisation with labelled probe
    + Requires some knowledge of RNA sequence
- Informative for Presence/Absence calls
    + Images scanned $\rightarrow$ *Densitometric Analysis* for crude quantitation
- Possible for some different isoforms to be detected
    + Sequence dependent

:::

## RT-qPCR

::: {.notes}
The C~T~ values is actually estimated to a decimal value
:::

- “Gold-standard” for measurement of transcription levels
    + Single gene $\implies$ not a high-throughput technique
- Targets a single transcript region with specific primers to produce cDNA <br>$\rightarrow$ Polymerase Chain Reaction (PCR)
- Each PCR cycle approximately doubles the target region
    
::: {.fragment}    
- cDNA produced is identified using fluorophores
    + Fluorescence doubles with each cycle
- Once fluorescence passes a detection threshold, the cycle number is recorded
    + Known as the Cycle Threshold (C~T~) value

:::

## RT-qPCR {.slide-only .unlisted}

![A 10-fold dilution series](transcriptomics/qPCR.png){fig-align="left"}

## RT-qPCR {.slide-only .unlisted}

- Higher C~T~ values $\implies$ lower numbers of target molecule at the beginning
- These can be used to estimate and compare abundance levels (i.e. gene expression)

::: {.fragment}
- Is vulnerable to technical artefacts (e.g. pipetting & sample variability)
- Often includes one or more "housekeeper" genes thought to be stably expressed
:::

::: {.incremental}
- C~T~ values are *normalised* to the housekeeper genes $\implies \Delta C_T$
    + log~2~ transformed values are used: $\Delta C_{T_g} = \log_2 C_{T_g} - \log_2 C_{T_{hk}}$
- Comparison between conditions is the change in $\Delta C_T \implies \Delta\Delta C_T$
- Represents change on the log~2~ scale, i.e. *log fold-change*
:::

## Expressed Sequence Tags

::: {.notes}
- The senior author on the EST paper was J Craig Ventner who played an important role in the Human Genome Project
:::

- The first attempt at capturing the larger transcriptome was ESTs [@1991VenterEST]
- Identified 609 human brain mRNA sequences
    + Selected for polyA-mRNA then reverse transcribed
    + Used random primers $\rightarrow$ Sanger Sequencing
- 10 years before the Human Genome Project
    + Gene discovery was a hot topic

## Sanger Sequencing

![Estevezj, [CC BY-SA 3.0](https://creativecommons.org/licenses/by-sa/3.0), via Wikimedia Commons](transcriptomics/Sanger-sequencing.svg){fig-align="left"}

## SAGE & CAGE

- First high-throughput quantification method was *Serial Analysis of Gene Expression* (SAGE) [@pmid7570003]

:::: {.columns}

::: {.column width="58%"}

::: {.incremental}
- mRNA $\rightarrow$ cDNA using biotinylated primers
- cDNA bound to beads (using biotin) & cleaved 
- 11mer "tags" were ligated into long sequenced using linker sequences
- Sequenced using Sanger Sequencing
- Deconvolution & counting
:::
:::

::: {.column width="42%"}
::: {style="font-size: 80%; text-align: center;"}
![Thomas Shafee, [CC BY 4.0](https://creativecommons.org/licenses/by/4.0), via Wikimedia Commons](transcriptomics/SAGE.png)
:::
:::

::::

## SAGE & CAGE {.slide-only .unlisted}

- The terminology of counting *tags* is still used by some manuals & software
    + Statistical models still form the basis of modern transcriptomics
- Was described as Digital Gene Expression (DGE)
    + The term DGE is still used but easily confused with Differential Gene Expression
    
::: {.fragment}
- A variant called Cap Analysis of Gene Expression (CAGE) targeted the 5' Cap
- Heavily used by FANTOM project [@Abugessaisa2021-fn] to identify exact Transcription Start Sites (TSS)

:::

# Microarray Technology

## Microarray Technology

::: {.notes}
- My search last week showed 69000 public microarray datasets in the GEO database
- I reviewed a Scientific Reports submission using public array data last month
:::

- Microarrays represent the *birth of modern transcriptomics*
    + Thousands of genes could be measured simultaneously!!!
    + Tens of thousands of public datasets $\implies$ still being mined
- Established during latter stages of the Human Genome Project (1990-2003)
    + Databases & complete reference sequences become widely available

::: {.incremental}
- All require fluorescently labelled cDNA copies of RNA
- Hybridised to the array using probes for known sequences
    + $\uparrow$ fluorescence $\implies \uparrow$ RNA abundance
:::

## Microarray Technology {.slide-only .unlisted}

- All microarrays follow the same basic process

![Image courtesy of Squidonius, Public domain, via Wikimedia Commons](transcriptomics/Microarray_exp_horizontal.svg)

## Two Colour Arrays

:::: {.columns}

::: {.column width="57%"}

- Two colour microarrays were printed microscope slides
- Known probe sequences were *printed* to the surface in defined locations
    + 60-75mer oligonucleotide probes
    + Highly customisable by project
    
::: {.fragment fragment-index=1}
- Two samples per array
    + Samples labelled with Cy5 (Red) or Cy3 (Green)
- Scanned at 570nm (Cy3) and 670nm (Cy5)

:::

:::

::: {.column width="43%"}

::: {.fragment fragment-index=1}

![Section of two-colour array taken from @Shalon01071996](transcriptomics/2colour.jpg){width="100%" fig-align="left"}

:::
:::

::::

## MA Plots

:::: {.columns}
::: {.column}
![Image from @dudoit02-ma](transcriptomics/dudoit_ma_plot.png)
:::
::: {.column}
- Mean of Differences<br>$M = \log_2(\frac{R}{G}) = \log_2(R) - \log_2(G)$
- Average Signal<br>$A = \frac{1}{2}\log_2(RG) = \frac{\log_2(R) + \log_2(G)}{2}$

::: {.fragment}
- Assess bias within and between arrays
- Also to show DE genes
:::
::: {.fragment}
- Term "MA Plot" still used in RNA-Seq despite no connection to formula
:::
:::
:::: 

## Single Channel Arrays

:::: {.columns}

::: {.column width="55%"}


- Affymetrix Arrays became dominant
    + Factory manufactured
- Standardised layout for each organism
- Single sample per array
    + Only scanned at one frequency<br>$\implies$ no dye bias
- More genes/array

::: {.fragment}
- 25mer probes targeting 3' end of transcript
    + Captured only intact transcripts
:::


:::

::: {.column width="45%"}
::: {style="font-size: 80%"}
![No author known. Schutz assumed based on copyright claims. [CC BY-SA 3.0](http://creativecommons.org/licenses/by-sa/3.0/), via Wikimedia Commons](transcriptomics/Affymetrix.jpg)
:::
:::

::::

## Single Channel Arrays  {.slide-only .unlisted}

:::: {.columns}

::: {.column width='85%'}

<br>

![The basic methodology underpinning Affymetrix array design. Source Affymetrix](transcriptomics/microarrayLayout.jpg){width="100%"}

:::
::::

## 3' Arrays

- Each 3' exon targeted by 11 unique 25mer probes $\implies$ *a probeset*
- Possible to detect different transcripts only if 3' exons differ

::: {.incremental}
- *Perfect Match* (PM) probes $\implies$ exactly matches target sequence
    + Known to capture off-target signal $\implies$ non-specific binding (NSB)
- 3' arrays include paired *mismatch* probes (MM) with a change at the 13th position
    + Literally half the array
    + Intended to quantify NSB properties of each probe
    + Sometimes returned more signal than PM probes &#129322;
<!-- - Gene-level expression estimate obtained taking a robust average across probeset -->
<!--     + Model also included BG signal (but not always from MM probes) -->
    
:::

## Whole Transcript Arrays



- Whole Transcript Arrays released by Affymetrix in mid-2000s
    + Marketed as Exon Arrays and Gene Arrays
- Probes along entire transcript *BUT* $\leq$ 4 probes/exon

::: {.fragment}

![Source: Affymetrix Technical Note](transcriptomics/ArrayProbeLayout.png){fig-align="left" width="70%"}
:::

## Whole Transcript Arrays  {.slide-only .unlisted}

::: {.notes}
- Detecting alternate isoform usage on Exon Arrays was the focus of my PhD
- My method was rubbish, but still better than anything else
:::

- No successful methods for determining alternate isoform usage
    + Most people reverted back to gene-level signal
    + No real gains over 3' Arrays beside more genes/array

::: {.fragment}

- RNA-Seq appeared at a similar time & destroyed Affymetrix's market share
    + Alternate isoform usage in RNA-seq is *still* considered a bit exploratory

:::

## Microarrays Vs RNA-Seq

:::: {.columns}
::: {.column}

```{r}
#| echo: false
#| message: false
#| fig-width: 6
#| fig-height: 6
library(tidyverse)
theme_set(theme_bw())
here::here("Lectures/transcriptomics") %>% 
  fs::dir_ls(glob = "*search.csv") %>% 
  lapply(read_csv) %>% 
  bind_rows(.id = "platform") %>% 
  mutate(
    platform = platform %>% 
      basename() %>% 
      str_remove_all("_search.csv") %>% 
      str_to_title() %>% 
      str_replace_all("Rna", "RNA-")
  ) %>% 
  ggplot(aes(Category, `Publications (total)`)) +
  geom_line(aes(colour = platform), linewidth = 1) +
  ggrepel::geom_label_repel(
    aes(label = label, colour = platform),
    data = . %>% 
      dplyr::filter(Category == max(Category), .by = platform) %>% 
      mutate(
        label = paste0(
          Category, "\n", scales::comma(`Publications (total)`))
      ),
    force_pull = 0, nudge_y = -1000,
    show.legend = FALSE, size = 4, fill = "#FFFFFF90"
  ) +
  geom_vline(xintercept = 2006, linetype = 2) +
  ggtitle("Publication search by app.dimension.ai on 26-08-2025") +
  labs(x = "Year", colour = "Platform") +
  scale_y_continuous(labels = scales::comma) +
  scale_x_continuous(
    breaks = seq(2000, as.integer(format(Sys.Date(), format = "%Y")) - 1, by = 5)
  ) +
  scale_colour_brewer(palette = "Set1") +
  theme(
    text = element_text(size = 16),
    plot.title = element_text(face = "italic", size = 14),
    legend.position = "inside",
    legend.position.inside = c(0.99, 0.01),
    legend.justification.inside = c(1, 0)
  )
```

:::
::: {.column}
::: {style="font-size:92%"}
- Exon Arrays released in 2006 (dashed line)
- Publications usually lag purchasing by 2-3 years
    + Microarrays peaked in 2008 (i.e. 2005-6)
    + Affymetrix owned by Thermo-Fisher since 2016
- Microarrays continue to be extensively used in DNA-methylation analysis

:::
:::
::::

# Microarray Analysis

## Single Channel Data

- We will have multiple arrays from each condition
    + Biological Replicates (hopefully $\geq4$ per condition)
    + Want to find changed expression in response to our biological hypothesis
    
::: {.incremental}
    
- Will some arrays have higher/lower overall signal?
    + Pipetting errors, hybridisation variability etc

- Two Initial Problems to solve

1. Adjust for overall differences in signal [$\implies$ Normalisation]{.fragment}
2. Removal of Background Signal (non-specific binding + optical noise) <br> [$\implies$Background Correction]{.fragment}

:::

## Normalisation

:::: {.columns}
::: {.column width="65%"}

![Example of raw PM probe intensities. Taken from @pmid12538238](transcriptomics/quantileNorm.png){fig-align="left"}
:::
::: {.column width="35%"}
- The variation here is primarily technical<br>$\implies$ *not* due to biology
- Higher variance reduces power of statistical testing
- Can we reduce this?

::: {.fragment}
- Quantile normalisation
:::
:::
::::

## Normalisation {.slide-only .unlisted}

- Quantile normalisation is perfect for arrays with probes and probesets
    + Normalise probe-level signal, but estimate gene expression at the probeset level
    + Smooths out any normalisation artefacts

::: {.incremental}
1. Select the lowest signal probe on each array<br>$\rightarrow$ Likely to be a different probe on each array
2. Calculate the average signal across all arrays
3. Give each of the selected probes the average signal
4. Move to the next lowest signal probe until finished
:::

::: {.fragment}
- Effectively randomises noise
- Leads to arrays with identical distributions
:::

## Normalisation {.slide-only .unlisted}

:::: {.columns}
::: {.column}
![Before quantile normalisation](transcriptomics/preNorm.png)
:::
::: {.column}
![After quantile normalisation[^3]](transcriptomics/postNorm.png)
:::
::::

::: {style="font-size:95%"}
- Now we have identical distributions of signal across all arrays
- Equivalent to having identical amounts of source material (mRNA)
- Reduces technical noise across dataset $\implies$ more statistical power
:::

[^3]: Images taken from: Bolstad, Probe Level Quantile Normalization for High Density Oligonucleotide Array Data Unpublished Manuscript, 2001

## Background Correction

- Background Correction performed *simultaneously with estimation of signal*
- Robust Multichip Average (RMA) [@rma2003]
    + Estimates signal for each array ($\mu_i$)
    + Model includes probe affinities ($\alpha_j$)
    + Doesn't include MM probes
    + Fitted using robust statistics to reduce impact of outlier probes
    
$$
\log_2 PM_{ij} = \mu_i + \alpha_j + \epsilon_{ij}
$$
    
::: {.fragment}

- Extended to GC-RMA [@wu2004model] to include GC content of probes

:::

## Differential Expression Analysis

- A primary challenge is to detect where gene expression levels change in response to biological question
    + Often control samples Vs treated samples
- Microarray data is normally distributed on the log~2~ scale
    + Can fit standard regression models


::: {.fragment}

$$
H_0: \text{No difference in average gene expression levels}\\
H_A: \text{Some difference in average gene expression levels}
$$

- NB: Experiments estimate the *true* expression level across a theoretical population
:::

## Differential Expression Analysis {.slide-only .unlisted}


$$
T = \frac{\beta}{\sigma/\sqrt{n}}
$$

- $\beta$ is estimate of effect size (i.e. logFC)

::: {.incremental}
- Some estimates for $\sigma$ are too low, others too high
    + Too low $\implies T \uparrow$ [$\implies$ significant result where no change]{.fragment}
    + Too high $\implies T\downarrow$ [$\implies$ no significant result where there is change]{.fragment}
    
:::

::: {.fragment}
- Variance estimates moderated by *taking distribution of* $\sigma$ *across all genes*
    + Bayesian posterior estimate of variance $\implies$ moderated t-statistic [@pmid16646809]
    
:::


## Differential Expression Analysis {.slide-only .unlisted}

- The Bioconductor package `limma` is the industry standard [@pmid16646809]
    + Still heavily used for modern RNA-Seq data
    + Models tailored to managing variances found in transcriptomic datasets

::: {.fragment}
- After testing $\rightarrow$ $p$-value for each gene
- Multiple testing becomes an issue (revise Steven Delean's lecture)
    + Mostly use the Benjamini-Hochberg FDR
    
:::

## Example Results

```
         logFC AveExpr     t  P.Value adj.P.Val     B
Gene 1   2.180  1.0834  7.96 4.78e-05   0.00478  2.52
Gene 34  1.361  0.1829  3.12 1.44e-02   0.53387 -3.47
Gene 2   1.980  1.3416  2.90 2.02e-02   0.53387 -3.82
Gene 72  1.127 -0.0684  2.86 2.15e-02   0.53387 -3.88
Gene 75 -0.524 -0.1168 -2.72 2.67e-02   0.53387 -4.10
Gene 86  0.771 -0.1801  2.52 3.60e-02   0.59375 -4.41
Gene 16 -0.501 -0.0495 -2.43 4.16e-02   0.59375 -4.55
Gene 83  0.453 -0.1065  2.27 5.35e-02   0.65833 -4.80
Gene 33 -0.932 -0.5752 -2.15 6.39e-02   0.65833 -4.97
Gene 89  0.467 -0.1272  2.13 6.58e-02   0.65833 -5.00
```

<br><br>*Taken from the* `limma` *help page examples*

## Volcano Plots

:::: {.columns}

::: {.column width="60%"}

![](transcriptomics/plot-volcano-1.png)

:::

::: {.column width="40%"}

- Show estimated of logFC against significance (-log~10~$p$)
- Actually taken from RNA-Seq but plots are identical

:::

::::

<!-- ## Multiple Testing -->

<!-- - Researchers commonly use a $p$-value $<0.05$ to claim evidence of significant change -->
<!--     + Often described as the rejection threshold $\alpha = 0.05$ -->

<!-- > A $p$-value represent the probability of observing data (i.e. evidence of change) as extreme, or more extreme than our data if $H_0$ is true -->

<!-- ::: {.fragment} -->

<!-- - i.e. how likely are we to see our evidence of change if nothing's actually happening -->
<!-- - Using $\alpha = 0.05$ we will randomly see evidence as strong as observed (or stronger) about 1 in 20 times, *if there's really nothing happening* -->

<!-- ::: -->

<!-- ## Multiple Testing {.slide-only .unlisted} -->

<!-- - If testing across 10,000 genes $\implies$ 10,000 $p$-values -->
<!-- - If no change between groups in any gene ($H_0$ is always true) <br>$\implies$ about 500 genes will return $p<0.05$ -->
<!-- - These would be considered *false discoveries*, *false positives* or *Type I Errors* -->
<!--     + False negatives are Type II errors -->


<!-- |              | $H_0$ True | $H_0$ False | -->
<!-- |:------------ | ---------- | ----------- | -->
<!-- | **Accept $H_0$** | &#x2705; | Type II Error | -->
<!-- | **Reject $H_0$** | [Type I Error]{.text-red} | &#x2705; | -->


<!-- ## Multiple Testing {.slide-only .unlisted} -->

<!-- - Reducing false positives is vital $\implies$ don't want to waste research \$\$\$ -->
<!-- - An intuitive solution would be to use a lower value for $\alpha$ -->
<!--     + Using $\alpha = 0.01$ $\implies$ 1 in 100 random $p$-values under $H_0$<br>$\implies$ only 100 false discoveries in 10,000 genes -->
<!--     + But this makes it harder for *true positives* to be identified <br>$\implies$ more Type II errors (false negatives) -->
<!-- - No perfect solution for minimising Type I errors without increasing Type II errors -->

<!-- ## The Bonferroni Correction -->

<!-- - A commonly used approach is the Bonferroni Correction -->
<!-- - $\alpha$ is divided by the number of tests <br>$\implies$ probability of one false discovery remains at the original $\alpha$ -->
<!--     + Known as Family Wise Error Rate (FWER) control -->

<!-- ::: {.incremental} -->
<!-- - For $n=10,000$ genes we would use $\alpha^* = \frac{0.05}{n} = 5\times10^{-6}$ -->
<!-- - The equivalent is to multiply all $p$-values by $n$ to give adjusted $p$-values -->
<!-- - $p_{adj} = \min(1, p \times n)$ ensures $p$-values are always $\leq1$ -->
<!--     + Referred to as Bonferroni-adjusted $p$-values -->

<!-- ::: -->

<!-- ## The False Discovery Rate -->

<!-- - An alternative is to allow a set proportion of your results to be *false positives* -->
<!-- - Instead of controlling the FWER, we control the False Discovery Rate (FDR) -->
<!--     + Using $\alpha = 0.05$ $\implies$ 95% of results will be appropriate rejections of $H_0$ -->
<!--     + Is an estimate of the true FDR -->
<!--     + Effectively allows a small amount of noise into our data -->
<!--     + For larger network-style or downstream analysis $\implies$ signal drowns out noise -->
<!-- - Is almost always more powerful that the Bonferroni correction -->
<!--     + More power $\implies$ more ability to find the *true positives* -->

<!-- ## The False Discovery Rate {.slide-only .unlisted} -->

<!-- ::: {.notes} -->
<!-- - The adjustment algorithm is a bit beyond this course -->
<!-- ::: -->

<!-- - Most common FDR method is the Benjamini-Hochberg adjustment [@BenjaminiHochbergFDR] -->

<!-- ::: {.incremental} -->
<!-- 1. $p$-values are given rank $i = 1, 2, \ldots, n$ where we have $n$ tests (i.e. genes) -->
<!-- 2. Find maximum value of $i$ where $p_i \leq \alpha \frac{i}{n}$ -->

<!-- - FDR-adjusted $p$-values easily obtained using `R` -->
<!--     + `p.adjust(p, method = "BH")` -->

<!-- ::: -->



## Enrichment Testing

- Often want to identify key pathways associated with results
    + Are genes from specific pathways or gene-sets most impacted

:::: {.columns}


::: {.column}
![](transcriptomics/apoptosisDAG.png){width="85%"}
:::


::: {.column}
- Gene Ontology (GO) Terms:
    + Carefully constructed, hierarchical database of terms
    + Biological Process, Molecular Function and Cellular Component
    
:::

::::


## Enrichment Testing {.slide-only .unlisted}

:::: {.columns}


::: {.column width="60%"}
![Oxidative Phosphorylation pathway as annotated in the KEGG database](transcriptomics/hsa00190.png){width="95%"}
:::


::: {.column width='40%'}
- Kyoto Encyclopedia of Genes and Genomes (KEGG):
    + Molecular pathways with complete topology
    
:::

::::

## Enrichment Testing {.slide-only .unlisted}

:::: {.columns}
::: {.column width="55%'}

- Often use approaches based on Fisher's Exact Test
    + Compare enrichment of DE genes in a pathway vs not DE Genes in pathway
- Pathways often share DE genes $\implies$ often need biological expertise
- Multiple Testing across 1000s of pathways

::: {.fragment}
- 20% of pathway is DE vs 6.7% of "not-pathway" genes
:::

:::
::: {.column width='45%'}
|     | DE Genes | Not DE Genes | %DE |
|:--- | --------:| ------------:| --:|
| In Pathway | 10 | 40 | 20% |
| Not In Pathway | 990 | 14000 | 6.7% |
:::
::::

## Enrichment Testing {.slide-only .unlisted}

::: {style='font-size:95%'}

:::: {.columns}
::: {.column width='45%'}
- An alternative approach is to use ranked lists
    + Gene Set Enrichment Analysis (GSEA)
- No requirement to classify genes as DE or not DE

::: {.fragment}
- Walk along the ranked list of genes increasing score if gene in gene-set
    + In the example, begin at the right
    + Largest deviation from zero is -ve

:::
:::
::: {.column width='55%'}
![Image taken from clusterProfiler vignette https://yulab-smu.github.io/clusterProfiler-book/chapter12.html](transcriptomics/gseaplot2-1.png)
:::
::::

:::


# Closing Summary

## Closing Comments

- Microarray signal estimates follow a normal distribution ($\mathcal{N}(\mu, \sigma)$) with log-transformed
- We can apply linear regression models
    + In a simple A vs B experiment $\equiv$ $T$-tests
- Estimates of change from DE analysis often referred to as logFC (log fold-change)
- Variances can be moderated for improved performance
- $p$-values are usually FDR-adjusted
    + Gives best compromise of power vs error-rate control

## Closing Comments {.slide-only .unlisted}

- Foundations built during the microarray era enabled analysis of RNA-Seq data
    + R/Bioconductor Community
- Core principles and methods developed during this era still apply
    + Normalisation, DE analysis, multiple testing, enrichment testing etc
- Many bioinformaticians from the microarray era are still very active
- A lot of development occurred in Australia (e.g. Prof Gordon Smyth, WEHI)
    + Next generation have been trained & mentored at WEHI, USyd etc
- RNA-Seq data is not normally distributed $\implies$ discrete counts not continuous fluorescence


## References
